\chapter{Centrality measures}




 \begin{displayquote}
 I asked a person of intelligence how many steps he thought it would take, and he said that it would require 100 intermediate persons, or more, to move from Nebraska to Sharon. \cite{milgram1967small}
 
 Stanley Milgram 1969
\end{displayquote}




 Networks can be studied at three levels. First the properties of the network as a whole, such as the small-world property, secondly the properties of individual nodes and their importance  within the network and finally the analysis of the presence of meso-scale structures such as communities.

This chapter examines measures of centrality with reference to the post synaptic proteome network and measures of intelligence. Centrality measures how important nodes are within a network \cite{newman2018networks} and their position within the network \cite{freeman1978centrality}. The concept of centrality comes from the social sciences literature beginning in the last 1940s and its early evolution is reviewed in Freeman \cite{freeman1978centrality}.


Essentially some nodes in a network can be more important than others such as an airport hub in an air transportation network.\cite{borenstein1989hubs} Centrality statistics provide measures of this importance which can allow nodes to be ranked or their centrality compared with other properties. This chapter describes these centrality measures, their previous use in the biomedical literature, reports their values and significance for the PSP, their implication for the function of the network and association with specific functions and their association finally with molecular cognition first in model organisms and finally with differences in cognitive ability in population GWA studies.   

 The importance of central genes in a network is well recognised \cite{jeong2001lethality} including their association with disease \todo{ref}.
 
 Important nodes in the network might disproportionately affect animal models of cognition or human differences in intelligence. One could hypothesise that central nodes will affect a number of genes and potentially have an impact on a pleitropic trait \todo{ref} such as intelligence. \todo{omnigenic model}.  Conversely there may be no such effect perhaps because effects on these critical nodes are deleterious and are either lethal or cause problems with cognitive ability that would absent sufferers from large scale genetic studies of cognitive ability. 
 
 Although network network science is new discipline it has been the subject of detailed popular science books \cite{barabasi2002linked} and one of its principles has been the subject of a play \cite{guare1990six} and film and also a parlour game \footnote{Six Degrees of Kevin Bacon}.\cite{collins1998s} In the light of such knowledge most educated persons would be able to give a more accurate answer to Milgram's question above. 

 
 
 Hindsight bias \cite{fischhoff1975hindsight} \cite{fischhoff2007early}, describes the phenomena where the probability of an event or finding seems greater after it has occurred (we know it all along)
 
Knowing the association between centrality measures and intelligence or any other complex trait one could construct a convincing argument that this finding is obvious and to be expected. However the specific effect of proteins position in the synaptic proteome network on synaptic function and complex traits are unknown. Even effects believed to be well established on simple model organisms are more complicated that they fist appeared as will be discussed in section \todo{cross reference}. The approach to assessing the role of central genes and gene products must therefore be empirical. 
 
 The existing literature on centrality measures in molecular biology include both \todo{rephrase} gene networks and this protein-protein interaction network. Network analyses of gene function and regulation are often co-expression networks where an edge occurs when the expression of two genes are correlated.\cite{zhang2005general} The implications of modular structure or centrality can, therefore, be different from a protein protein interaction network. A hub node is co-expressed with a variety of different genes and may represent a driver of a biological process such as a transcriptional regulator \cite{parikshak2015systems}\todo{ DONE cite parikshak}. A hub node in a protein-protein interaction network has promiscuous interaction partners but may not drive a process but be essential to several different processes. \todo{cite}
 
% \textcolor{green}{? remove  this bitor move to earlier bit}While the measures of vertex importance are precisely defined how they will map onto animal models or complex traits is not and one can construct several plausible hypotheses of effect, effect of specific measures or no effect (hence the quotation at the start of the chapter\todo{?move this down into a fuller introduction section}.

\section{General to dos}

Table of diameters \cite{crescenzi2013computing}
 table 4 
 \todo{enrichment of high/low scoring groups of genes}
 \todo{add intellectual disability graph measures or indeed other disorders from toppgene}
 \todo{define configuration model}
 \todo{plot of distribution of other centrality measures}
\todo{centrality measures for significant genes}
\todo{centralisation 1:09 in zhukov - centrality is how central most central node is compared to other nodes in graph, max with star network}
\todo{Metrics comparison zhukov at 1:15}
%Scatter plot of centralities multiplot eg plot(bet,deg) DONE

Kendal tau rank - concordant pairs ie a comes before b in both samples 

most recent centrality in \url{source('~/RProjects/centrality/R/other_centrality/eccentricity_and_distance.R')}



\section{Properties of empirical networks}
\label{sec:Properties of empirical networks}
In addition to community structure which we will examine in chapter~\ref{chap:community detection} empirical networks may contain features at a network level such as a power law degree distribution \cite{barabasi1999emergence} \cite{barabasi1999mean} and small world structure \cite{watts1998collective}. These can provide information on how networks form and evolve.\todo{ref} We have that advantage of knowing several facts about the evolution of the PSP importantly the gene duplication event \cite{grant2016molecular} They are not all shared by all empirical networks. For example road networks do not show a scale free property although air travel networks do.\textcolor{red}{? remove this or move it into the bit above}\todo{? remove this section or merge with earlier}. We can test for the existence of these properties by determining how a network differs from either a completely random model or a model in which some of its features (such as degree distribution (section \ref{sec:degree distribution}).
\todo{?add global graph measures}

\section{Small world}
\label{sec:Small world}
\textcolor{red}{this is not covered in introduction}
Described by Milgram \cite{milgram1967small} where he attempted to see the number of steps that it would take for a letter to reach a reciepient unknown to the first letter holder if they could only mail it to people known personally to themselves. The average number of links was 7. This has been made into a popular play by John Guare. Watts and Strogatz \cite{watts1998collective} described small world properties in naturally occuring networks (nerves of c elegans, power network and film actor network) which differed from a random model. These properties are a short average path length with high clustering coefficient. 

Watts and Strogatz \cite{watts1998collective} suggest a generative model for this phenomenon with a circular lattice network in which interconnections are rewired to span the circle. A comparatively small number of rewirings allows a short path length and preserves the high clustering coefficient. The small number needed has been cited as one of the reasons it is common in nature. 

The small world property would allow for rapid information transfer within networks. It has recently been posited as a model for wide spread genetic effects on traits (the omnigenic model) \cite{boyle2017expanded} although there are criticisms of this some of which I will make later \todo{reword}.\textcolor{green}{what i mean so say is there are some cirticisms of the omnigenic model saying that things are more complex eg \cite{wray2018common} \cite{boyle2017omnigenic_response}  but the criticism I am making is from a network standpoint} 


\todo{small world network either do here or put in introduction this goes with results }
. Results see section~\ref{sec:Results average path length and transitivity}

\subsection{The configuration model}
\label{sec:configuration model}
\todo{do this}
In considering measures of  a network or a vertex we can study them by seeing how much they differ from their expected value given a random network. Simple random networks such as Erdos Renyi random graphs are discussed in the introduction \todo{cross ref}.

These models however result in a Poisson distribution of node degree (see section~\ref{sec:degree distribution}) \todo{move this to after degree distribution} and many real world networks have more complicated degree structures such as a power law distribution (see section~\ref{sec:scale_free}) . A number of generating models for power law distribution networks have been described. If one is studying a particular network for which one has the degree sequence then the configuration model is a good choice \todo{rephrase}.

The configuration model keeps the degree distribution of the network. Each edge is broken into two stumps and then the stumps are rewired into other stumps with a uniform probability. This can result in self loops and multiedges which do not occur in most real world networks but the number of these are low as the network size increases \cite{newman2018networks}.




\section{Centrality measures}
Centrality measures show how important as node is within the network \todo{said this before} and a variety of statistics use different measures of importance (see section~\ref{sec: intro_centrality_measures}). Many are correlated with each other although their correlation varies.\textcolor{red}{move this down}. We will consider degree, eigenvector centrality, betweenness, closeness and transitivity. All measures are derived from the adjacency matrix.



Other vertex measures include eigenvector centrality, a measure of the importance of the node that includes the degree of its neighbours. \cite{bonacich1987power}  Some nodes are bottlenecks in the network. If a large number of the shortest (geodesic) paths pass through a node it is a potential bottleneck and its removal can disrupt the integrity of the network. The measure of a nodes propensity to be a bottleneck is the betweenness centrality. \cite{freeman1977set} 

Figure~\ref{fig:dolphin} shows vertex betweenness centrality, eigenvector centrality and local transitivity. High values of these measures are shown in red. The data is from a social network of bottle nose dolphins \footnote{\url{http://www-personal.umich.edu/~mejn/netdata/dolphins.zip}} \cite{lusseau2003bottlenose} \todo{Make vertices bigger in figure}.

\begin{figure}
    \centering
    \includegraphics[width=0.9\textwidth]{images/centrality2.002.png}
    \caption{Display of centrality measures. Betweenness centrality, eigenvector centrality and local transitivity for network of bottle nose dolphin social relations data. \cite{lusseau2003bottlenose}}
    \label{fig:dolphin}
\end{figure}

The simplest centrality measure is degree centrality considered below. 

\section{Degree centrality}
\label{sec:degree}

Probably the simplest centrality measure is degree. 
%from \url{/home/grant/Dropbox/PhD_latex_master}
The degree of a node is the number of edges connected to it. In the network graph of the PSP it is the number of proteins a particular protein interacts with.
For the adjacency matrix $A_{i,j}$, where $A_{i,j}=1$  if there is an edge between nodes $i$ and $j$ and 0 otherwise, the degree of node $i$, $k_i$ is\cite{boccaletti2006complex}:
\begin{equation}
k_i = \sum_{j=1}^n A_{i,j}.
\label{Equation:Degree_from_adjacency}
\end{equation}

 Nodes with high degree may have more influence in a graph and are often referred to as hubs. Note however there are two senses in which the term hubs may be used (see p 168 Newman book\cite{newman2018networks}).  In a social network  individuals with a large number of contacts will have high degree and hence be influential in the network. In a protein protein interaction network high degree proteins will have numerous interactions and one would hypothesise play an important part in the network. 
 
 Disruption of high degree nodes leads to disproportionate effects upon a network \cite{jeong2001lethality} \cite{albert2000error} 
\todo{Sort out removed citations because of undefined control}
\todo{also include evidence against yeast thing from dominating sets paper}

\subsection{Degree distribution}
\label{sec:degree distribution}
The degree distribution of a network is the probability that a node chosen at random has a particular degree or the empirical probability density of the network eg

\begin{equation}
    p(k) = k
\end{equation}

or can be represented as a degree sequence which is the actual instantiation of node degrees for each vertex in a network eg $1,1,2,3,5,7,\dots,n$. The properties of networks can be discovered by comparing these networks to random models. 
If a graph were to have a random distribution of edges between vertices we would expect the degree distribution ($P(k)=k)$) to be distributed according to the binomial distribution.  

If there is a constant probability $p$ of an edge between two nodes in a network, the probability of having $k$ links with probability of links between vertices of $p$ is $p^k$ and the remaining $N-1-k$ nodes being absent (-1 as we exclude self loops) is $(1-p)^{N-1-k}$.

There are $\binom{N-1}{k}$ ways in which this arrangement of present and missing edges can be arranged so the probability follows the binomial distribution.

\begin{equation}
   \binom{N-1}{k}        p^k (1-p)^{N-1-k}
   \label{Equation:BinomialDistributionForDegreeProbability}
\end{equation}
In the limit where $E[k] << N$ the degree is well approximated by a Poisson distribution \cite{barabasi2016network}. The Posson distribution is parameterised only by $E[k]$.  The probability mass should be centred around the empirical mean and we would not expect to see nodes whose degree were 3 x or more the average degree. This is not what we find the distribution is scale free. \todo{probablility of max degree node} 
\todo{plot of Poisson and empirical for $E[k]$} \textcolor{red}{see figure}


\subsection{Scale free}
\label{sec:scale_free}
\todo{define scale free}
\subsubsection{Description of scale free}

\textcolor{red}{need to give background on barabasi etc}
\todo{background scale free}
It has been recognitsed that the degree distribution of nodes in real world networks are commonly scale free \cite{barabasi1999emergence} \cite{barabasi1999mean}. In a scale free network the probability of a vertex having degree $k$ $p(k)$ is:
\begin{equation}
    p(k) \sim k^{-\gamma},
\end{equation}
\label{eq:scale free}

where $\gamma$ is the degree exponent \cite{barabasi2016network}

Rare \cite{broido2019scale}

\textcolor{red}{Why called this scale free}




\section{Value of tissue specific networks}
\cite{parikshak2015systems} quotes the example of cardiac PPI to investigaste long QT in \cite{lundby2014annotation}

Kitsak and Barabasi \cite{kitsak2016tissue}

\subsection{Comparison of degree distribution to random erdos graph}
See section~\ref{sec: intro_random_graphs}
\todo{Comparison of degree to random graph with same p}

\begin{equation}
    <m> = \binom{n}{2}p
\end{equation}


\begin{equation}
    c = (n-1)p
\end{equation}
where c is the mean degree see newman p 346

Add a histogram here and the expected mean and quartiles.


\subsection{Degree and giant component}
\label{sec:connected component and degree}
Some structure can develop even in random graphs see~\ref{sec: intro_random_graphs}
\todo{Change this as below subcritical, supercritical, fully connected} CHANGE this
In a random graph there is a point at which scattered connected nodes combined to form a giant component. Subcritical to supercritical at $<k> >1$


Supercritical to fully connected at $E[k] > \ln(N)$


Where $N$ is the number of nodes in a network, a giant connected component develops in random graphs when $E[k] > \ln (N)$ \cite{barabasi2016network}. This tends to be the case in real world complex networks. The necessary requirement for a connected component to form is that mean degree is greater than $1/N$. The giant connected component arises in the super-critical' region (see \cite{barabasi2016network} p87). For the PSP $E[k]$ is 17.6 and $\ln(N)=\ln(3457)=8.14$. \todo{this does not follow} It is important to note however that the degree distribution for the PSP graph \textcolor{red}{this does not follow} is not that of a random network ($P(k)$ is scale free it would also be unlikely to find many nodes $ 3 \times E[k]$, if it were there would be few nodes with degree 2 and we find some nodes disconnected from the connected component in our empirical model \todo{so what?}. \todo{cross ref to missing nodes I think it is in the paper section so community detection} \todo{Add plot from \url{'~/RProjects/graph_analysis/R/transitiontoconnectederdosrenyi.R'} for transition from sub-critical to super-critical region} see section~\ref{sec: PSP graph connected component and missing} for missing components.


\subsection{Degree and intelligence in PSP}
We will see what the correlation is in population cohorts and what we might expect it to be given the effects of high degree genes in model animals. 
\todo{This is perhaps the bit to introduce the possible different hypothesise}


%%% CENTRALITY MEASURES %%%
\section{Other centrality measures}

\subsection{Betweenness}
\label{sec:Betweeness centrality}
Both betweenness centrality and closeness centrality (section~\ref{sec:closenesscentrality}) depend on the properties of shortest paths between nodes. Betweenness centrality is a vertex level statistic described by Freeman \cite{freeman1977set}. Betweenness centrality is a measure of information flow through a node and the importance a node measured by the number of geodesic (shortest) paths between nodes in the network passing through a particular node. 

Betweenness centrality measures the number of shortest (geodesic) paths through a node. Following Newman if $n_{st}^i$ is one if a shortest path between nodes $s$ and $t$ pass through node $i$ and $g_{st}$ is the number of shortest paths between $s$ and $t$ then the betweenness centrality of node $i$ is

\begin{equation}
    x_i = \sum_{st} \frac{n_{st}^i}{g_{st}}
\end{equation}
\label{eq: Betweenness centrality}

from Newman \cite{newman2018networks} p172. 

Informally the number of shortest paths passing through a node for all paths between nodes in the network and where more than one shortest path occurs it is weighted by the number of such paths that exist. The betweenness centrality can be normalised by dividing by the number of possible node pairs in the network $n^2$ \cite{newman2018networks} p176

Nodes with high betweenness centrality connect different areas on the network (ie paths between nodes in the network disproportionatley flow through them) they are sites of increased information flow (especially if the information knows the path to take through the network (compare with random walk betweenness centrality) and measures how much other nodes depend on the presence of these nodes.


\textcolor{red}{there is a note betweennness and transitivity and betweeness and biological illegible maybe literature}
 
\subsubsection{Edge betweenness}
 \todo{edge centrality}


\section{Closeness centrality}
\label{sec:closenesscentrality}
Alex Bevelas

Closeness centrality measures how centrally ie close to other nodes a node is in the network. Like betweenness centality it depends on shortest walks. Closeness centrality is the inverse of the  average length of shortest path to and from other vertices in the graph \cite{freeman1978centrality}

\begin{equation}
    l_i = \frac{1}{n} \sum_j d_{ij}
\end{equation}

$l_i$ is the mean distance between node $i$ and every other node in the network. As a distance measure it is small when nodes are central so we take its inverse to produce closeness centrality


\begin{equation}
    C_i = \frac{1}{l_i}
\end{equation}

While betweenness centrality measures the dependence of other nodes in the network on a particular node (as a bottleneck for information) the closeness centrality has been interpreted both as proximity and access efficiency and independendce from other nodes \todo{ cite(Bronelles, Brandes and Borgetti)} but these are identical



\section{Eigenvector centrality}

Eigenvector centrality proposed by Bonacich \cite{bonacich1972factoring} expands on degree centrality by not considering all of the edges of a vertex equal. If a vertex is linked to a highly important node this link should be given greater importance. Eigenvector centrality therefore incorporates the importance of the neighbours of a vertex.

Here I follow Newman \cite{newman2018networks}. From the above we define a nodes importance as the sum of the importance of its neighbours.

\begin{equation}
    x_i = K^{-1} \sum_j x_j
\end{equation}

Where $K$ is a constant of proportionality. This can be written using the adjacency matrix as

\begin{equation}
    x_i = K^{-1} \sum_j A_{ij} x_j
\end{equation}

Where $\sum_j x_j$ includes the importance of all nodes but selects only the neighbours of $x_i$ because of the adjacency matrix (otherwise the entry will be 0).

In matrix form:
\begin{equation}
    \mathbf{x}=K^{-1}\mathbf{Ax}
    \end{equation}
    
    \begin{equation}
         k\mathbf{x}=\mathbf{Ax}
    \end{equation}
  
  Replacing $k$ with $\lambda$ shows more clearly this to be an eigenvector problem. The eigenvector will have as its $i$th entry the importance of the $i$th node. As all centralities should be positive the appropriate eigenvector is the one associated with the largest eigenvalue by the Perron-Froebenius theorem. 
  
  Eigenvector centrality is implemented in igraph for R by the \texttt{eigen\_centrality} function taking the graph as the argument. The values of the eigenvector centrality are not strict measures and we care only for the order of eigenvector centraliy. They can however be normalised. 
  
 \section{Related measures}
\cite{bonacich1987power}
Nodes are important if their neighbours are important 

Katz centrality has a parameter (alpha)
    
Alpha centrality and bonacich centrality have parameters (bonacich is alpha with alpha preset per bonacich). Bonacich is power\_centrality in igraph. Power centrality with preset is symmetrical around 0.  

\todo{? add pagerank}




\section{Transitivity}
\label{sec:transitivity}
The transitivity of a network measures the extent to which if two nodes each connected to another node they will be in turn connected. In more colloquial terms if Alice and Bob are both friends with Susan how likely are Alice and Bob to be friends. The clustering coefficient measures the number of these connections called closed triads occurring in a network 
divided by the possible number that could occur.

Transitivity can be calculated for the network as a whole or for individual nodes. This global transitivity is often called the clustering coefficient which can be confusing because of the alternate clustering coefficient described below in section~\ref{sec:Global clustering coefficient}. 

A triangle of vertices,  all connected to each other, forming a loop is  a \textit{closed triad} \cite{newman2018networks}. It is also a path of length two (as paths must have distinct edges. The possible triads that could be formed by the addition of an edge are all paths of length two so the transitivity is:

\begin{equation}
    C = \frac{\textrm{number of closed paths of length two}}{\textrm{number of paths of length 2}}
\end{equation}

Since there are six paths that can be present in a closed triad (as we count each direction the path can take in an undirected network eg $ab$ and $ba$ are distinct the transitivity is:

\begin{equation}
    C = \frac{6 \times \textrm{number of triangles}}{\textrm{number of paths of length 2}}
\end{equation}

Using connected triple to refer to any connected three vertices (whether they form a loop or not) the transitivity can then be expressed as.

\begin{equation}
    C = \frac{6 \times \textrm{number of triangles}}{2 \times \textrm{number of connected triples}}
\end{equation}
The probability the two neighbours of a node in a network are connected is thus its transitivity.\cite{newman2018networks}. 

\todo{expected value}


\subsubsection{Global clustering coefficient}
\label{sec:Global clustering coefficient}
The local transitivity defined below (section \ref{sec:local clustering coefficient} is used synonymously with clustering coefficient. The global clustering coefficient however defined by Watts and Strogatz as the arithmetic mean of the vertex local clustering coefficients. The global clustering coefficient can become domminated by nodes of low degree \cite{newman2018networks} p188 and nodes with low degree tend to have high values of local clsutering coeffieicnt. 

\subsection{Local clustering coefficient}
\label{sec:local clustering coefficient}
Local transitivity (also known as the local clustering coefficient) is defined as  follows \todo{Citation error}cite(watts1998collective) for a vertex having $k_v$ neighbours 

\begin{equation}
C = \frac{c_n}{c_{max}}
\end{equation}
where $c_n$ is the number of connections between a vertexes neighbours and $c_{max}$ is defined as 

\begin{equation}
c_{max} = \frac{k_v(1-k_v)}{2}
\end{equation}

The local clustering coeffiecient can be regarded as a local inverse of the betweenness centrality \todo{check}. Local clustering measures the connectivity of nodes in the neighbourhood of a particular node. If the local clustering is low then there exist disconnections in a nodes neighbours which are referred to as structural holes \todo{cite ?Burt} the node that connects these neighbours has an importance analagous to that measured by betweenness centrality but confined to the immediate neighnourhood of the node. .

\todo{power}





\subsection{kcoreness}

A kcore of a graph is an induced subgraph of the main graph such that each vertex has degree of at least $k$. For a vertex its kcoreness is the maximal k core subgraph it is a member of (alternatively its shell index is $c$ if it belongs to the $c$ core but not the $c+1$ core. \cite{seidman1983network} (original description), \cite{alvarez2006large}. The kcore algorithm is useful in graph analysis as it runs in linear time. 

\subsection{Average path length}
\label{sec:Centrality intro average path length}

\section{Assortativity and assortativity of measures}
before editing
Assortativity is a measure of how likely nodes with similar properties are to be connected. \cite{newman2002assortative} by either scalar or categorical vertex properties. The most commonly studied scalar property is degree association. Assortative networks tend to form communities and are more robust to disruption by node removal  \cite{newman2002assortative} and increased assortativity leads to increased speed of transmission of information through a graph \cite{noldus2015assortativity}. \todo{plot of knng and deg at \url{source('~/RProjects/correlation_gene_scores/R/make_df_graph_stats_correlation_lowdeg_high_bet.R')}} although there seems to be a very weak correlation between knn and z score. 
\subsection{Degree assortativity}
\label{sec:degree assortativity}
\subsubsection{Introduction Degree assortativity}
The degree assortativity is the Pearson correlation between degree for connected vertices.\cite{noldus2015assortativity} Explain assortativity
Newman p 208
Equation assortativity


\subsection{Other measures}
\subsubsection{Shortest paths}


\subsubsection{Bridging}

\cite{valente2010bridging}



\section{Biomedical significance of centrality measures}

 High degree nodes are called hubs and proteome hub proteins are known to be more commonly essential to life \cite{jeong2001lethality} and expressed ubiquitously. \cite{goh2007human}  Jeong found that high degree nodes in yeast were more likely to be essential to life \cite{jeong2001lethality} although later work has questioned this (see section\todo{cross ref} Unregulated\todo{check unregulated or upregulated} genes in cancer proteomic networks have been found to be of high degree. \cite{wachi2005interactome}  High degree nodes were  involved in more diseases in the Online Mendelian Inheritance in Man (OMIM) compendium of genetic phenotypes. \cite{xu2006discovering}  Goh however found that disease genes were likely to be tissue specific and were not more likely to be hubs. \cite{goh2007human}  
 
High degree hub genes are more likely to be essential rather than disease genes \cite{barabasi2011network}. 

Lee et al. \cite{lee2013network} in 2013 examined National Human Genome Research Institute (NHGRI)  catalogue of complex trait GWAS \footnote{now the NHGRI-EBI GWAS catalogue} and found that complex trait loci were more likely to be found in genes that were hubs or bottlenecks, where hub and bottleneck are defined as the top 20\% of the degree centrality and betweenness centrality distribution.  The network is composed of the genes encoding protein protein interactions in the STRING (14025 nodes and 492087 edges representing protein protein interactions). The GWAS datasets were composed of type 2 diabetes studies (FUSION  Cases n=	1161 Controls n=	1174 
WTCCC Cases n=	2000 controls n=	3000  and Inflammatory Bowel Disease Genetics Consortium (IBDGC) with European populations without Jewish ancestry and those with Jewish ancestry (IBDGCi case n=	561 control n=	563 IBDGCii case n= 	407 control n=	432).  

Liu \cite{liu2017sigmod} described a method of identifying gene modules and combining this with p values but use heterogenous different measures of gene interaction including coexpression from the STRING database version 10 which results in what they refer to as a GeneNet \todo{there is a thing called a gene net - check} with 19 247 genes and 4 274 001 edges. Edges are given weights based on level of evidence but if there is a connection but no weighting a value of 1 is given. Nodes are given higher significance based on degree.  Gene level P values were obtained by calculating the value of the top SNP within a gene. They compare their results to two other methods dmGWAS and SCoNEs. Their method is implemented in an R module (SigMod) last updated 2 years ago \todo{remove this this is to jog my memory}. They appear to use a min cut algorithm to find modules and take into account degree. THere is also a complex method for adjusting two tunable parameters \todo{can't really make head or tail of this paper. Don't know why they didn't use something like VEGAS. }



\cite{bayes2011characterization} found strong sequence conservation in PSP hub proteins. So there is some evidence that we might suspect they have an influence on human cognition using the G2C framework but as this is complex what we need is empirical proof indeed one can make a plausible argument from the available evidence for the effect of a specific measure, for an effect of centrality generally or for no effect \todo{? add box}.  


\subsection{ Biomedical Review}
Ghasemi provides a review of centrality measures in biological networks as of 2014 and illustations of each centrality measure on Zacharys Karate network.
\cite{ghasemi2014centrality}. Some of the papers cited are discussed below but the reader is directed to Ghasemi \cite{ghasemi2014centrality} for a fuller review. 

Hahn and Kern\cite{hahn2005comparative} examined protein protein interactions networks in fly, yeast and worm measuring betweennness centrality, degree centrality and closeness centrality. They found that more centrally located proteins in all networks evolved more slowly. 

Ozgur \cite{ozgur2008identifying} developed a curated gene interaction network and used centrality measures to attempt to predict the genes association with prostatic cancer using the prostate gene database (PGDB). They measured degree, eigenvector, betweenness and closeness centrality and found that high scores on the first three measures were increased in increased confidence in rank in the PDGB.

Estrada \cite{estrada2006virtual} used centrality meassures to assess essentialness in the yeast protein interaction network (degree centrlity, eigenvector centrality, information centrality, closeness centrality, betweenness centrality and subgraph centrality. They found that high centrality measures predicted essentialness better than random selection and  the best results were obtained using eigenvector centrality and the poorest using closeness and betweenness centrality. 

Joy found that nodes with high betweenness and low degree centrality were particularly abundant.\cite{joy2005high} (this does not seem to be the case in our data see \url{lm_1 <- lm_1 <- lm(ZSTAT ~  eig*scale_bet*I(1/scale_deg),data=ukbb_int_df2)}

Pandey analysed GWAS results for bipolar disorder using eigenvector centrality \cite{pandey2012epistasis}. They used the WTCCC-BD GWAS (1868 cases and 2938 controls) as a discovery sample and  a NIMH GWAS (1001 cases; n=1033 controls) of individual of european ancestry as replication. Having constructed an epistasis network they used eigenvector centrality to rank the 1000 nodes in the SNP epistasis network and then assessed what pathways were enriched for the ranked SNPs. The analysis is limited by the lack of power provided by the sample size available.

\section{Degree and essentialness Introduction}
\label{sec:Degree and essentialness}
Jeong \cite{jeong2001lethality} examined the correlation between a gene being essential and having high degree in the proteome of \textit{Saccharomyces ceresvisiae} having 1870 nodes and 2240 interactions (comparatively sparse compared with the synaptic proteome). They found that elimination of the most connected nodes led to an increase in network diameter and that the likelihood that mutation in a protein would prove fatal was correlated with degree, They found that 21\% of the nodes with degree less than 5 (93\% of the proteome). Although 0.7\% of yeasts with more than 15 links had phenotypic profile 62\% of these were lethal. However later work with more complete networks did not support this \cite{milenkovic2011dominating} \cite{yu2008high}


Raman \cite{raman2014organisational} examined multiple centrality metrics in addition to degree (degree, betweenness centrality, closeness centrality and pairwise dis-connectivity index) and their association with gene essentiality for 20 organisms.

 Using the STRING database filtered for high quality interactions they found a disassortative \todo{ this suggests you need to put assortativity in the introduction bit} pattern of connection of essential genes (they did not appear to be connected to each other)  (compare with section~\ref{sec:degree assortativity})  and found higher than average degree and betweenness centrality (greatest for degree centrality). The organisms that they studied (20) were mostly bacteria and yeasts, no most sophisticated I could see was \textit{Caenorhabditis elegans}. They also calculated what fraction of essential nodes had a higher degree than n where n is a centile of the degree distribution).

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    METHODS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{METHODS}

\subsubsection{Methods scale free property}
\todo{define the method of Clauset and Newman}
\todo{intro to scale free}
We can calculate the $\gamma$ coefficient for the degree distribution using the \texttt{poweRlaw} package for R. The value of $\gamma$ will vary over the distribution depending on the lower limit of degree used to fit the power law distribution. This is shown graphically in figure \ref{fig:gamma} and table \ref{table:gamma}.


\section{Gene ontology enrichment}
\textcolor{red}{Cross ref to bit in next section or bring it here}

To understand the properties of nodes of different centralities we performed gene ontology enrichment both with the background of the PSP and against the whole of the genome where the whole of the genome is defined as genes appearing in at least one of the cohort studies \todo{add cross ref}

\subsection{GO Slim}
\textcolor{red}{Define}

\subsection{Methods and results global clustering coefficient}
Calculations of local transitivity were carried out using the transitivity method in igraph with the argument type being global undirected. Local transitivity was carried out using transitivity with the type being local transitivity is not defined for nodes of degree 0 so as an alternate to NA the argument isolates="zero" was used. 

\subsection{Methods Murine LTP and centrality}
Identification of murine genes associated with LTP and testing of LTP and cnetrality. For results see section \ref{sec:results centrality and murine models of long term potentiation}.

%%%%%    Methods essentialness %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Database of essential genes Methods}
 \label{sec:Database of essential genes}
 
 
 The database of essential genes is found at \url{www.essentialgene.org}. Downloading essential eukaryote genes yields 34590 entries.
 
 Data is available for 9 clades:  Saccharomyces cerevisiae   ,Caenorhabditis elegans,Arabidopsis thaliana,Danio rerio,Mus musculus,Homo sapiens,
 Drosophila melanogaster,Aspergillus fumigatus and Schizosaccharomyces pombe 972h-.
 
 Data on 28286 \todo{is this correct this is more than the number of protein coding genes?} human genes are recorded. 8256 unique gene ids are found for human genes. Human genes are identified by HGNC id code \footnote{\url{source('~/RProjects/paper_xls_latex/R/essentail_genes/load_essential_genes.R')}}.
 
 Entrez id were obtained using the module \url{convert_gmx_private}\footnote{ at \url{/home/grant/PycharmProjects/convert_gmx_private}}.
 
 This uses downloaded gene info from entrez gene to find first the symbol and if that is missing to search through synonyms and supply the matching entrez id. It returns a list of items not found. 
 
8284 genes were translated to entrez id. 8 ids could not be identified. 3 of these are recorded as dates in the primary data \footnote{(5,6 Mar, 5 Sep)}
 
 The dataframe of conversions are stored in the supplemental data\footnote{ \url{/home/grant/RProjects/paper_xls_latex/data/essential_genes.tsv}}
\subsection{Loss of function and evolutionary pressure}

The Exac database is a compendium of data from the exomes of over 60,000 individuals \cite{lek2016analysis}. It permits the calculation of the frequency of mutations in specific genes. In addition there is a background mutation rate that is observed in human genes. If there is a lack of mutations in a specific region then we can conclude that the organism is intolerant to such mutations. The researchers provide a statistic the probability of loss of function intolerance that provides a measure of how much purifying selection a gene is under. The file was downloaded from 
\url{"https://storage.googleapis.com/gnomad-public/legacy/exac_browser/forweb_cleaned_exac_r03_march16_z_data_pLI_CNV-final.txt.gz"}

\subsubsection{Processing and cleaning Exac data for PSP}
\todo{query move to supplemental methods}
The data consists of 18282 rows and 23 columns. A variety of metrics are available to measure the frequency of synonymous mutations, non synonymous mutations and missense mutations. The authors recommend the use of the probability of loss of function mutation metric as this takes into account variations in gene size which some other metrics may be biased by. In addition the rows were identified by Ensembl transcript ID, HUGO gene symbol and contained information on the number of base pairs, chromosome number and location on the chromosome. This data is recorded in supplementary section~
\ref{sec:supplemental data table for exac}.


% raw output for ref
%[1] "gene"       "chr"        "n_exons"    "tx_start"   "tx_end"    
% [7] "bp"         "mu_syn"     "mu_mis"     "mu_lof"     "n_syn"      %"n_mis"     
%[13] "n_lof"      "exp_syn"    "exp_mis"    "exp_lof"    "syn_z"      "mis_z"     
%[19] "lof_z"      "pLI"        "n_cnv"      "exp_cnv"    "cnv_z"     
% end

114 duplicated entries with duplicated Ensembl transcript id were identified. Each was duplicated twice with identical transcript id and entrez gene ID (duplicate of 57 entries) see table~\ref{tab:Supplementary table- duplicate exac data}. The duplicates were identical other than different numbers of copy number variations (CNV) recorded. As the analysis was not going to include CNV these were discarded.  \textcolor{red}{this matches the redo}\footnote{code \url{('~/Rprojects/paper_xls_output/R/download_exac.R')}}.


Entrez ID were obtained by joining the Exac data to the human gene info 

% The genes were identifiable from both transcript ID ENST, and HGNC gene symbol. (18282 - 57) 18225 distinct transcripts and gene symbols were identified.

% 57 transcript id and gene ID were duplicated and removed  There were 114 duplicated entries (i.e. each was duplicated twice). For all duplicates all statistics were identical of than number of copy number variations, expected number of copy number variations and z score for copy number variations. 
% \todo{move into separate function}

% As we were not using CNV these were eliminated

To obtain entrez ID we join the Exac data to the human gene info downloaded from entrez. This joined table has 8 duplicates as a result of Exac gene symbols mapping to more than one entrez ID)

One synaptic gene is found in the duplicated list HBD Gene ID 3045 which also has the entrez ID 100187828 
HBD
the transcript is ENST00000380299.3 which is haemaglobin delta maps to ENSG00000223609

there is also Hypophosphataemic bone disease which is 100187827 but the one that is present in the synapse is 3045


Code for the extraction and processing of the loss of function intolerance data was as follows 
\url{~/RProjects/paper_xls_output/R/download_exac.R}

3235 genes of the PSP have exac data. The missing 222 have insufficient coverage. The missing ones are shown in supplemental table x.
Lookup on \url{https://grch37.ensembl.org/Multi/Search/Results?q=ENST00000404774.3;site=ensembl_all}

MEMO1 mediator of cell motility Entrez 7795 (Methylation modifier for class I HLA
) and 51072(correct)

MMD2 monocyte to macrophage 221938 MMD2 100505381 (Miyoshi Muscular Dystrophy 2 – wrong)

TEC tec Protein tyrosine kinase 7006 not 100124696 (Transient erythroblastopaenia of childhood)

There are 14990 non synaptic genes with exac information and a total of 18225 \textcolor{red}{18070 in redo} genes with exac gene info



\subsection{METHODS Network statistics from paper}
\label{sec: network statistics from paper}

To test the hypothesis that measures of gene importance in the network are correlated with cognitive ability and/or educational attainment, we calculated a range of standard network measures: betweenness centrality, degree, clustering coefficient, closeness centrality and eigenvector centrality using igraph for Python 3.6. \cite{csardi2006igraph}  We calculated the correlation between each of these vertex scores and the -log10 transform of the gene level p values derived from MAGMA for each study using Spearman’s rank correlation using the Scipy stats package for Python 3.6. \section{Measures of vertex importance and intelligence:Methods}

\label{sec:Measurs of vertex importance from paper}
\todo{Move this to importance}
To test the hypothesis that genes encoding proteins with an important role in the structure of the PSP network were more likely to be associated with genetic variants associated with differences in intelligence or educational attainment (such as hubs), we calculated node degree, and four other measures of vertex importance: eigenvector centrality, betweenness centrality, closeness centrality and transitivity for each graph vertex (gene) using igraph. \cite{csardi2006igraph} \textcolor{red}{we need to say what all of these are before we start talking about them}\footnote{something like we will first of all need to introduce in general terms these measures and how they were calculated. in general also want to move more of this chapter into methods and results}. There was no correlation between any of these measures of vertex importance and their gene association analysis gene level statistics calculated using MAGMA (Spearman’s $\rho$ calculated with R) for genes in the post synaptic proteome graph. The results are presented in table 1\todo{table}. The complete vertex statistics for each PSP gene can be found in the supplementary material. 

\section{Cohorts and samples}
\label{Centrality:cohorts and samples}
\todo{do correlation centrality with murine ltp group}
The population data used to test the association of centrality measures with population studies of educationbal attainment and intelligence were those described in sections~\ref{sec:cohorts from paper section} and \ref{sec:samples from paper section}

\todo{Add introduction}


%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%    RESULTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\section{RESULTS}
? divide into graph level and vertex level
\todo{? divide into graph level and vertex level}

\section{Summary of centrality measures}

The summary values for the centrality measures are shown in table \ref{Table:Summary of centrality measures}. The transitivity contained 304 NA's.
\todo{add graph}
The eigenvector centrality and transitivity are normalised. For some nodes the transitivity cannot be calculated (those of degree 1). There are 304 of these and the summary statistics are calculated excluding these values\footnote{an alternative is to set them to zero}. 

The centrality measures are not normally distributed see table~\ref{Table:Test for normality (Shapiro-Wilk) for centrality measures of PSP}.



\footnote{\url{source('~/RProjects/paper_xls_latex/R/centrality_summary/centrality_summary_latex.R')}}
% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Mon Feb 10 14:17:12 2020
\begin{table}[ht]
\centering
\begin{tabular}{rrrrrrr}
  \hline
 & Minimum & 1st Quartile & Median & Mean & 3rd Quartile & Maximum \\ 
  \hline
Degree & $1 $ & $4 $ & $8 $ & $17.64$  & $19$ & $535$ \\ 
  Eigenvector & $1.821 \times 10^{-6}$ & $8.829 \times 10^{-3}$ & $2.194 \times 10^{-2}$ & $4.793 \times 10^{-2}$ & $5.322 \times 10^{-2}$ & $1 $ \\ 
  Betweenness & $0 $ & $43.29 $ & $317 $ & $3421.1$ & $1571.6$& $6.447 \times 10^{5}$ \\ 
  Transitivity & $0 $ & $4.376 \times 10^{-2}$ & $1.290 \times 10^{-1}$ & $1.714 \times 10^{-1}$ & $2.381 \times 10^{-1}$ & $1 $ \\ 
  Closeness & $5.702 \times 10^{-5}$ & $9.217 \times 10^{-5}$ & $9.879 \times 10^{-5}$ & $9.845 \times 10^{-5}$ & $1.058 \times 10^{-4}$ & $1.399 \times 10^{-4}$ \\ 
   \hline
\end{tabular}
\caption{Summary of centrality measures} 
\label{Table:Summary of centrality measures}
\end{table}
\todo{Global transitivity and compare}
\todo{query pull out the global measures eg degree assortativity, transitivity}


\begin{table}[h]
    \centering
    \begin{tabular}{c|c|c}
       Centrality measure  &  W& p\\
       \hline
       
       Degree  & 0.426 & $<2.2 \times 10^{-16}$ \\
       Eigenvector &0.561  & $<2.2 \times 10^{-16}$ \\
       Betweenness &0.1228& $<2.2 \times 10^{-16}$ \\
       Transitivity &0.782 & $<2.2 \times 10^{-16}$\\
       Closeness &0.9953& $<4.686 \times 10^{-09}$\\ 
    \end{tabular}
    \caption{Test for normality (Shapiro-Wilk) for centrality measures of PSP}
    \label{Table:Test for normality (Shapiro-Wilk) for centrality measures of PSP}
\end{table}

\section{Correlation of centrality measures}

A scatterplot of the relation of centrality measures is shown in figure~\ref{fig:scatter plot of multiple centrality measures}.

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{images/Rplot_pairs_plot.png}
    \caption{Scatter plot of centrality measures. Source \url{source('~/RProjects/correlation_gene_scores/R/make_df_graph_stats_correlation_lowdeg_high_bet.R')}}
    \label{fig:scatter plot of multiple centrality measures}
\end{figure}

\section{ Results Average path length}
\label{sec:Results average path length and transitivity}
The average path length calculated using the \texttt{mean\_distance} function in igraph is 2.98


This is similar to the L of 2.65 and C of 0.28 of C elegans \todo{reference and what is l and c I think c is clustering coefficient}. The random rewiring with a uniform probability leads to a L of 3.11. \todo{add table of path length}
\todo{add omnigenic bit}%see code source('~/RProjects/graph_transitivity/R/transitivity.R')

If we simulate mean distance using a degree sequence model we get no distance $<$ empirical distance of 2.979 in 1000 simulations.
\subsection{Results The characterisation of high degree nodes}
The first question we must ask of high degree nodes what are they. Is there something it is like to be a high degree node so we might expected difference behaviour of the nodes in disorders.
\todo{Do we therefore do this for all centrality measures or remove it or just degree}
Are nodes of high degree different to the rest of the PSP in terms of function. 
\textcolor{red}{Why are you telling me this that there is something it is like to be a high degree node so we might expect different enrichment in disease. I think it is high degree nodes are different from a random sampling of the PSP however they do not affect this specific phenotype the literature on centralities is contradictory and has to go piece by piece and with best evidence which in this case is tissue specific networks and GWA}

% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan  4 12:23:24 2020
\begin{table}[ht]
\centering
\scalebox{0.8}{
\begin{tabular}{rrll}
  \hline
Entrez & Degree & Symbol & Description \\ 
\hline
351 & 535 & APP & amyloid beta precursor protein \\ 
  2335 & 474 & FN1 & fibronectin 1\footnote{Fibronectin lethality} \\ 
  1994 & 473 & ELAVL1 & ELAV like RNA binding protein 1 \\ 
  1956 & 450 & EGFR & epidermal growth factor receptor \\ 
  7514 & 396 & XPO1 & exportin 1 \\ 
  10482 & 375 & NXF1 & nuclear RNA export factor 1 \\ 
  7316 & 363 & UBC & ubiquitin C \\ 
  26270 & 342 & FBXO6 & F-box protein 6 \\ 
  7412 & 327 & VCAM1 & vascular cell adhesion molecule 1 \\ 
  55832 & 316 & CAND1 & cullin associated and neddylation dissociated 1 \\ 
  8454 & 316 & CUL1 & cullin 1 \\ 
  2885 & 313 & GRB2 & growth factor receptor bound protein 2 \\ 
  7534 & 298 & YWHAZ & tyrosine 3-monooxygenase tryptophan 5-monooxygenase \\
  3320 & 266 & HSP90AA1 & heat shock protein 90 alpha family class A member 1 \\ 
  10075 & 263 & HUWE1 & HECT, UBA and WWE domain containing 1, E3 ubiquitin protein ligase \\ 
  4869 & 263 & NPM1 & nucleophosmin 1 \\ 
  7415 & 245 & VCP & valosin containing protein \\ 
  3178 & 227 & HNRNPA1 & heterogeneous nuclear ribonucleoprotein A1 \\ 
  22938 & 221 & SNW1 & SNW domain containing 1 \\ 
  8453 & 221 & CUL2 & cullin 2 \\ 
   \hline
\end{tabular}
}
\caption{The 20 post synaptic proteome genes with highest degree in their cognate proteins.NOTE tablewidthmodifiedusingscalebox. Code to generate table at \url{source('~/RProjects/gridsearch_gamma/R/degree_distribution.R')}} 
\label{Table:The 20 post synaptic proteome genes with highest degree in their cognate proteins.}
\end{table}
Table \ref{Table:The 20 post synaptic proteome genes with highest degree in their cognate proteins.} shows the 20 post synaptic proteome genes with highest degree and their associated degree. 7 of these genes are part of the ubiquitin protein binding process. \todo{Add class analysis panther}


\begin{figure}
    \includegraphics[width=12cm]{Rplot_DegreeDistribution}
    \caption{Degree distribution of post synaptic proteome. Log10 - log10 scale. Code at \url{source('~/RProjects/gridsearch_gamma/R/plot_log_degreedist.R') }}
    \label{fig:Degree distribution of post synaptic proteome. Log10 - log10 scale.}
\end{figure}

\subsection{Degree and Gene Ontology enrichment}

Using the 95th centile of degree distribution (59) we obtained 165 genes. Gene ontology enrichment analysis shows that they are over represented compared with the rest of the proteome for tau protein binding, ubiquitin pathways and cadherin binding.

\subsection{Enrichment compared to PSP}

Using SLIM ontology enrichment was found compared to the rest of the PSP for the biological process (table~\ref{tab:GO SLIM ontology enrichment Biological Process background PSP})

\begin{table}
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{llllllllll}
Name & Ontology ref &ref &	expected &	Fold Enrichment &	+/-	&raw P value&FDR\\
\hline
ubiquitin protein ligase & GO:0031625&  	7 &  	.74 & 	9.48 &  	+ & 	$5.08\times 10^{-05}$ & 	$2.43 \times 10^{-02}$\\
\end{tabular}
\end{adjustbox}
\caption{GO SLIM ontology enrichment Biological Process background PSP \textcolor{red}{PUT THESE ALL INTO ONE TABLE}}
\label{tab:GO SLIM ontology enrichment Biological Process background PSP}
\end{table}



SLIM Biological process was enriched for ubiquitin protein ligase (table~\ref{tab:GO SLIM ontology enrichment Biological Process background PSP})% latex table generated in R 3.6.3 by xtable 1.8-4 package
% Tue Jun  2 15:25:40 2020






\todo{redo some of these to check}
\begin{table}
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{l l l l l l l l l l}
Name & Ontologyref &ref & set&	expected &	Fold Enrichment &	+/-	&raw P value&FDR\\
\hline
proteasome-mediated ubiquitin-dependent &GO:0043161 & 48 & 	12 &	2.73 &	4.40 & 	+ & 	6.27E-05 & 	4.98E-02 \\
protein catabolic process\\
cell cycle& GO:0007049&	175 	&26 &	9.94& 	2.61 &	+ & 	3.29E-05& 5.22E-02\\
\end{tabular}
\end{adjustbox}
\caption{GO SLIM Biological process High degree nodes}
\label{tab: high degree slim biological process}
\end{table}

\subsubsection{Cellular compartment GO SLIM}
GO SLIM cellular compartment revealed enrichment for intra-nuclear organelles (table~\ref{tab: high degree slim cellular compartment}).

\begin{table}
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{llllllll}
  Name  &ref &set &	expected &	Fold Enrichment &	+/-	&raw P value&FDR\\
\hline
nucleus &	459& 	50& 	26.08& 	1.92& 	+ &	$1.60\times 10^{-05}$& 	$6.67\times 10^{-03}$\\
intracellular membrane-bounded organelle&	489&	51&	27.79&	1.84&	+&	$4.21\times10^{-05}$ &	$8.78\times 10^{-03}$\\
intracellular organelle&	490&	51&	27.84&	1.83&	+&	$4.31\times 10^{-05}$&	$5.98\times10^{-03}$\\
\end{tabular}
\end{adjustbox}
\caption{High degree GO SLIM cellular compartment}
\label{tab: high degree slim cellular compartment}
\end{table}

\subsubsection{Molecular function full}
Ontology enrichment of the full molecular
 function revealed enrichment of the WNT signolosome too
\begin{table}
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{l l l l l l l l}
Name  &ref & set&	expected &	Fold Enrichment &	+/-	&rawPvalue&FDR\\
\hline
Wnt signalosome& 	7& 	5& 	.40& 	12.57&	+ &	$2.51\time10^{-04}$& 	$8.15\times10^{-03}$\\ 
\end{tabular}
\end{adjustbox}
\caption{Molecular function. Full set ie not go slim ? PSP}
\label{tab:Molecular function. Full set ie not go slim ? PSP}
\end{table}

5 of the 7 elements of the signalosome are in the core segment and there are only 12 in the genome altogether.

\subsection{GO SLIM enrichment against all genome} 

\subsubsection{Cellular compartment }
See table~\ref{tab:Cellular component high degree GO SLIM against genome}
\begin{table}
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{lllllllll}
  Name & GO&	Homo sapiens &ref &	expected &	Fold Enrichment &	+/-	&raw P value&FDR\\
  COP9 signalosome&GO:0008180&	13&	3&	.16&	18.35&	+&	9.41E-04&	2.81E-02\\
  microtubule cytoskeleton &GO:0015630&	335&	13&	4.21&	3.09&	+&	4.40E-04&	1.64E-02\\
\end{tabular}
\end{adjustbox}
\caption{Cellular component high degree GO SLIM against genome}
\label{tab:Cellular component high degree GO SLIM against genome}
\end{table}

\subsubsection{Molecular function}
See table~\ref{tab:Molecular function high degree GO SLIM against genome}
\begin{table}
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{lllllllll}
  Name & GO&	Homo sapiens &ref &	expected &	Fold Enrichment &	+/-	&rawPvalue&FDR\\
\hline
ubiquitin protein ligase binding&GO:0031625&	34&	7&	.43&	16.37&	+&	6.59E-07&	1.66E-04\\
\end{tabular}
\end{adjustbox}
\caption{Molecular function high degree GO SLIM against genome}
\label{tab:Molecular function high degree GO SLIM against genome}
\end{table}

\subsubsection{Biological process}
Biological process bit of mixed bag nothing interesting 

\footnote{text files in \url{/home/grant/RProjects/gridsearch_gamma}}

\section{Gene ontology R topgo}
\subsection{Degree}

% latex table generated in R 3.6.3 by xtable 1.8-4 package
% Tue Jun  2 14:39:06 2020

\subsubsection{Results betweenness centrality Statistics and GO enrichment}
The log transform of betweenness centrality is more nearly normal. The histogram is slightly right skewed.
Although the qq plot looks relatively straight the Shapiro Wilk test for normality after removing infinite values and NA of the log transform  is 
W = 0.997 p = $1.1 \times 10^{-5}$

Top 142 betweeness
Panther 2016

Parkinson’s disease p 2.48 x 10-22
Dopamine mediated signalling pathway

Mammalian phenotype – lethality
BP – regulation of mRNA stability 
MF kinase binding and ubiquitin
CC cell adhesion
	]
\subsubsection{Results Edge betweenness results Statistics and GO enrichment}
The highest edge betweeness (21221.7 Median 321,4 mean 583.7) is between APP (351) and EGFR 1956.

The second highest is GRB2 (2885) and APP.
APP is on 12 of the top 20 edges. 
ELAV1 is on 5 of the top 20 edges.see \url{source('~/RProjects/centrality/R/other_centrality/eccentricity_and_distance.R')} EGFR (1956), 2885 and 7514 appear three times.

\section{Results:Closeness}
Closeness centrality \todo{add this} does not follow a power law distribution per p331-332 of Newman as it has a limited distribution as the reciprocal of the shortest path length from a node to all other nodes. 
\subsection{Results GO enrichment Closeness}
More normally distributed.
High closeness RNA binding, severe murine phenotype, cellular component cell adhesion
Ca and neurodegenerative

Low closeness centrality transporter activity, seizures in human phenotype, receptor activation, disease top five all epilepsies. 

\subsection{Results global clustering coefficient}
Transitivity is calculated locally and globally. The local transitivity has been calculated in igraph for R for a histogram of values see figure~\ref{fig:transitivity}. The local clustering coefficient has been calculated using graph tool. 
The global clustering as defined by Watts and Strogatz as the arithmetic mean of the vertex local clustering coefficients is 0.156.\cite{watts1998collective} this is the value if you set NA ie degree 1 to 0 and 0.171 is the value using NA.rm see above section. 

The expected value of local clustering given a random rewiring with fixed degree distribution is 0.07479691 sd (0.00183) removing NA and 0.068 setting NA to 0 (100 iterations) and 0.0682 seeting NA to zero with sd 0.00181 \footnote{Code  \url{source('~/RProjects/centrality/R/transitivity/calculate_local_transitivity_and_simulate_mean.R')} }



The correlation of the of the degree and local\_clustering (isolates set to 0)  is 
	Spearman's rank correlation rho

%data:  clustering_sub\$log_clustering and clustering_sub\$log_degree
S = 3670800000, p-value $< 2.2 \times 10^{-16}$
      rho= 
-0.373025 

Spearman's rho calculated as graph not linear throughout relationship see fig~\ref{fig:transitivity}.

\textcolor{red}{The literature MOVE TO DISCUSSION} records a degrees in the average value of the clustering coefficient with increasing $k$. This has been done mostly for large networks such as the internet \cite{newman2018networks} p 335. If we plot the degree and clustering coefficient there appears to be a negative correlation (figure~\ref{fig:Scatter plot of the relationship between degree and local transitivity for the PSP nodes}). The degree is not normally distributed so we have calculated Spearman's rank correlation. The transitivity with isolates set to NA and NA removed from the correlation is 0.196 p $< 2.2 \times 10^{-16}$. With the option of setting isolates to 0 the correlation is higher with $\rho$=0.375. This is however due to the large number of nodes of degree 1 (hence transitivity 0) and the low transitivity of degree two nodes. For degree greater than or equal to 2 the transitivity is 0.196. Greater or equal to 3 0.056 with p=0.0024 and equal to or greater than 5 rho is -0.085 with p= 3.02 $\times 10^{-15}$. If we plot the minimum degree to calculate the correlation and plot
this against rho calculated at that degree we can see the pattern clearly (figure~\ref{fig:Plot of minumum degree included in the calculation of Spearmans rank correlation between degree and rho}. The plot is truncated a degree 450 as after this point there are only two data points .
This relationship between $k$ and the transitivity $C$ therefore probably holds true in large networks (such as the internet where the studies were performed) where the number of nodes of low degree is dominated by the number of high degree (in this case $>=5$ nodes). Median node degree is 8, 5 corresponds \textcolor{red}{ to the 0.369 quantile}.

Reviewing the previous literature on the correlation between degree and local clustering Ravasz and Barabasi in a highly cited work \cite{ravasz2002hierarchical} review a generative model of high clustering coefficient and scale free degree distribution and real world networks. The real world networks include the actor network 392340 nodes, the language network based on synonyms in Merriams English dictionary 182853 nodes, the world wide web from mapping out from the \url{www.nd.edu} domain with 325729 nodes, the internet at the router level 260657 nodes and the power grid of the Western United states 4941 nodes 13188 edges. \textcolor{red}{end move to discussion}

These networks are all large and in the logarithmic graphs of clustering coefficient and degree the minumum degree starts around 10. It is interesting to note in figure 4 showing the internet at router level and power network there are lower degree nodes but the slope of the line representing the correlation coefficient si dominated by the effect of high degree nodes and the distribution of clustering coefficient in the lower degree domain is quite different. These matters may seem small as for large degree the overall pattern holds but it is a useful practical fact to be aware of if we are testing the association of clustering coefficient with another variable in smaller networks such as the PSP which is not readily apparent from the literature.

(Although in measuring the correlation of transitivty with another measure you are still measuring its correlation but a high transitivity low degree region may have different behaviour in the network to the majority degree greater than 10 negative correlation with transitivity)\\todo{correlation both demains}

code \url{source('~/RProjects/centrality/R/transitivity/calculate_correlation_min_transitivity_and_plot.R', echo=TRUE)}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{images/Rplot02_degree_and_transitivity-change_alpha.png}
    \caption{Scatter plot of the relationship between degree and local transitivity for the PSP nodes. Nodes with degree 1 have transitivity set to zero as transitivity in this case is undefined. Plot appears to show a negative correlation between degree and transitivity. The opacity of points has been decreased to reduce overplotting and show the increase in points at low degree}
    \label{fig:Scatter plot of the relationship between degree and local transitivity for the PSP nodes}
\end{figure}
\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{images/Rplot01_min_degree_transitivity.png}
    \caption{Plot of minumum degree included in the calculation of Spearmans rank correlation between degree and $\rho$. Although the overall coefficient is positive the correlation for nodes above degree 5 are clearly negative and this becomes more apparent at high degree. This relationship is therefore complex and the typical relationship described in the literature may only hold for the large networks in which this relationship has been studies such as the internet. }
    \label{fig:Plot of minumum degree included in the calculation of Spearmans rank correlation between degree and rho}
\end{figure}

The global undirected transitivity of the PSP network is 0.0697. This is similar to the value reported in p305 Newman \cite{newman2018networks} for metabolic networks (0.090) and PPI (0.072)\todo{check both different global formRplotulae in Newman}.\todo{describe method}

The expected value of the global clustering coefficient with a given degree distribution connected randomly \cite{newman2018networks} p332 is:
\begin{equation}
    C=\frac{1}{n}\frac{[<k^2> - <k>]^2}{<k^3>}
\end{equation}

The value of this for the PSP is 0.004541763 \footnote{code \url{source('~/RProjects/centrality/R/transitivity/Expected_global_transitivity.R')}}. The global clustering coefficient is therefore high. Newman states p334 \cite{newman2018networks} that it is not clear why certain non social networks have higher than expected global clustering (for social networks it is ascribed to social choice). For other networks such as the internet with very right skewed degree distribution the clustering coefficient is less than expected. For biological systems such as food webs or the world wide web it is hypothesised that this may due to community formation (newman ref 367)

Also motifs (other than number of triangles) newman 334 again

\subsubsection{Local clustering coefficient Results}
Local clustering coefficient tends to be inversely proportional to degree. It is undefined for nodes of degree one. See~figure \ref{fig:log_transitivity_degree}. It is suggested that this may also be due to community formation \cite{newman2018networks} p335.

Mean local transitivity is 0.171. The distribution is right skewed with median of 0.129 and 3rd quartile of 0.23 and first quartile of 0.043. This does not match the arithmetic mean of vertex local clustering below it may be down to how we deal with NA. Above calculated as NA.rm
\subsubsection{Results GO enrichment local clustering coefficient}

There was little enrichment for local clustering coefficient using GO and 0.05 and 0.95 centile in part because of the large number of 0 values when isolates are treated as 0. However in table~\ref{tab:Number of nodes 935 local transitivity 0.05 centile  local transitivity $<=$ 0 CC background PSP.Alpha=3.33555703802535e-05isolates 0} we can see that low local transitivity i.e. transitivity equal to 0 nodes (isolates) are enriched for plasma membrane components, the plasma membrane components do not show marked transitivity which is perhaps to be expected as they are at the edge of the PSP and it would not be intuitive for these components to all be interconnected. 


\subsection{kcoreness results}
 The minimum shell index is 1, all of the 1 kcore corresponds to the connected component. The range of shell index is 1-24. The median shell index of a vertex is 8 and the mean value 9.15 (although shell indexes take on integer values). The distribution of these values is shown in figure~\ref{fig:Kcore_histogram}\todo{nice ish image of kcore already in core periphery section}
 
 \begin{figure}
     \centering
     \includegraphics[width=\textwidth]{images/Rplot01_kcore_hist.png}
     \caption{Histogram of shell index of vertices in calculation of kcore. Shell index is the maximum kcore a vertex is a member of and not a member of shell index n+1. }
     \label{fig:Kcore_histogram}
 \end{figure}
 
 
 \subsubsection{K core gene ontology}
 
 Gene ontology enrichment for molecular  a range over the outer core 1-7 is shown in table~\ref{tab:kcore range GO Number of nodes 1724 k core 0 centile  k core1-7 MF background PSP. Alpha=3.33555703802535e-05}. It shows enrichment for trans-membrane transporters and G protein coupled receptors. 
 
 The innermost core $k>=23$ shows enrichment for RNA processing as seen in other centrality measures (table~\ref{tab:Number of nodes 201 k core 0.95 centile  k core $>=$ 23 BP background PSP. Alpha = 4.5583006655119e-06})

Code for GO and centrality measures \url{source('~/RProjects/group_sizes/R/plotting/modified_example...'} 
 
 

\section{Correlation of centrality measures}

For correlation of centrality measures see table~\ref{tab:Correlation between vertex centrality measures for PSP. Spearman's rho}


% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Fri Feb 14 12:11:44 2020
\begin{table}[ht]
\centering
\begin{tabular}{rrrrrr}
  \hline
 & degree & betweenness & transitivity & closeness & kcoreness \\ 
  \hline
degree & 1.000 & 0.902 & 0.196 & 0.851 & 0.986 \\ 
  betweenness & 0.902 & 1.000 & $-0.002^*$ & 0.782 & 0.854 \\ 
  transitivity & 0.196 & $-0.002^*$ & 1.000 & 0.292 & 0.265 \\ 
  closeness & 0.851 & 0.782 & 0.292 & 1.000 & 0.872 \\  
 

  kcoreness & 0.986 & 0.854 & 0.265 & 0.872 & 1.000 \\ 
   \hline
\end{tabular}
\caption{Correlation between vertex centrality measures for the post synaptic proteome network. Spearman's rho. All $p < 2.2 x 10^{-16}$ other than * $p=0.90$} 
\label{tab:Correlation between vertex centrality measures for PSP. Spearman's rho}
\end{table}




All of the centrality measures are correlated with each other other except transitivity and betweenness centrality  (table~\ref{tab:Correlation between vertex centrality measures for PSP. Spearman's rho}). Transitivity is most weakly correlated with other measures. Code \footnote{\url{source('~/RProjects/paper_xls_output/R/make_df_correlation_between_centrality_measures.R')}}
\todo{move comparison here}



See \cite{oldham2019consistency} for correlation across networks. See table~\ref{tab:Correlation of centrality valente et al}

\subsection{Results of correlation analysis FROM PAPER}
\textcolor{red}{Moved from earlier}
The vertex measures are correlated with one another apart from transitivity and range from 0.89 (eigenvector centrality and degree) to 0.33 (betweenness centrality and closeness centrality) (\textcolor{red}{see supplementary table 6} \todo{table}).
There is a weak negative ($\rho \approx 0.2$) correlation between 
the degree of selection pressure a gene is under and all measures of vertex importance except for transitivity (\textcolor{red}{see table 7 supplemental material} \todo{table}).
Although nodes of high importance or local clustering do have distinct properties, for example nodes with degree $> 50$ are enriched in association for the biological process (GO:1903350)‚ ”intracellular response to dopamine‚” (FDR P= 2.10 $\times 10^{-4}$ fold change 22.37), we find no correlation with these measures and genetic association with educational attainment or intelligence. 
\textcolor{green}{DRAFT -something like This is not because nodes of high degree or other centrality measure do not share common attributes, for example nodes with degree $>50$ are disproportionately found in BP intracellular dopamine but we find no association between these measures and educational attainment or intelligence}



\subsection{RESULTS Scale free}
Using the method described by \cite{clauset2009power} and calculating the minimum x value we find the min for $x$ to be 24 and the $\gamma$ coefficient to be 2.51 using the poweRlaw R package \cite{gillespie2015fitting}. GOF 0.19. ntail 68 \footnote{\url{source('~/RProjects/group_size_distribution/R/calulate_cdf.R')}} The log of the degree distribution is shown in figure \ref{fig:log_degree_distribution} and figure \ref{fig:Degree distribution of post synaptic proteome. Log10 - log10 scale.}
% see http://127.0.0.1:21133/library/poweRlaw/doc/a_introduction.pdf
\footnote{code at \url{/home/grant/RProjects/PhD_graphs/}}


The bootstrap p for the power law distribution is 0.583 with goodness of fit 0.19 using the package poweRlaw 0.70.6 using ks distance.\footnote{\url{source('~/RProjects/group_size_distribution/R/calulate_cdf.R')}}

To determine the best distribution calculated the log likelihood of the distribution given power law, log normal and poisson distribution. To allow comparison of the log likelihood the likelihood was calculated for the data from the $x_min$ determined for the powerlaw distribution $x_min = 24$. See table~\ref{tab:log likelihood powelaw}

\begin{table}[]
    \centering
    \begin{tabular}{cc}
    Distribution     &  log likelihood \\
    \hline
     Power law    & -2943.6\\
     Log normal & -2947.23 \\
     Poisson & -205115.8 \\
    \end{tabular}
    \caption{Log likelihood of degree distribution using poweRlaw package with $x_{min}=24$ for power law distribution and log normal and poisson.\url{source('~/RProjects/group_size_distribution/R/side_by_side_lognorm_and_pl.R')}}
    \label{tab:log likelihood powelaw}
\end{table}

\begin{figure}
    \centering
    \includegraphics[width=\linewidth]{images/Rplot01_poweRlaw_ggplot.png}
    \caption{Plot of fit of power law to degree distribution for the PSP. X axis is log 10 degree, Y axis is log 10 inverse cdf. Red line fit of power law distribution $\gamma= 2.51$ and $x_{min}=24$ \url{source('~/RProjects/group_size_distribution/R/calulate_cdf.R')}}
    \label{fig:poweRlaw plot ggplot2}
\end{figure}




\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{images/Rplot_poisson_plotted_on_degreedistribution.png}
    \caption{Plot of degree distribution PSP with degree on x axis (log10). Points are degree distribution. Blue dash expected degree distribution given Poisson distribution. Red dash average degree \url{source('~/RProjects/group_size_distribution/R/poissonanddegree.R')}}
    \label{fig:PSP degree powerlaw poisson}
\end{figure}

% \begin{figure}
%   \includegraphics[width=\linewidth]{./dependence_of_scale_coefficient_on_min_degree.png}
%   \caption{The parameter $\gamma$ as a function of the degree at which calculation of the coefficient starts in the degree sequence(e.g. all degree $>5$) from \url{RProjects/PhDGraphs/calculate/gamma}} 
%   \label{fig:gamma}
% \end{figure}

% \begin{figure}
%   \includegraphics[width=\linewidth]{plot_log_degree_distribution.png}
%   \caption{The plot of log cumulative cdf of degree distribution log 10 scale x and y}
%   \label{fig:log_degree_distribution}
% \end{figure}


\begin{figure}
    \centering
    \begin{subfigure}[t]{0.45\textwidth}
        \centering
        \includegraphics[width=\linewidth]{images/dependence_of_scale_coefficient_on_min_degree.png} 
        \caption{The parameter $\gamma$ as a function of the degree at which calculation of the coefficient starts in the degree sequence(e.g. all degree $>5$) from \url{RProjects/PhDGraphs/calculate/gamma}} \label{fig:gamma}
    \end{subfigure}
    \hfill
    \begin{subfigure}[t]{0.45\textwidth}
        \centering
        \includegraphics[width=\linewidth]{images/plot_log_degree_distribution.png} 
        \caption{The plot of log cumulative cdf of degree distribution log 10 scale x and y} \label{fig:log_degree_distribution}
    \end{subfigure}
    \caption{Plots of degree distribution \textcolor{red}{may want to do this in subfigure packages rather than subcaption to line up edges}}
\end{figure}
% latex table generated in R 3.4.4 by xtable 1.8-4 package
% Sat Oct  5 16:31:0\begin{document}

\begin{table}[ht]
\centering
\begin{tabular}{lr}
  \hline
  from degree $k$ & coefficient $\gamma$ \\ 
  \hline
1 & 1.38 \\ 
  5 & 1.81 \\ 
  10 & 2.10 \\ 
  15 & 2.30 \\ 
  20 & 2.43 \\ 
  25 & 2.51 \\ 
   30 & 2.52 \\ 
  35 & 2.54 \\ 
   40 & 2.54 \\ 
   45 & 2.60 \\ 
  50 & 2.63 \\ 
   60 & 2.60 \\ 
   70 & 2.72 \\ 
   80 & 2.80 \\ 
   90 & 2.90 \\ 
   100 & 2.92 \\ 
  150 & 2.88 \\ 
   200 & 3.44 \\ 
   250 & 3.90 \\ 
  300 & 5.04 \\ 
   350 & 5.51 \\ 
   400 & 6.33 \\ 
   \hline
\end{tabular}
\caption{The $\gamma$ as a function of the degree sequence start (e.g. all degree $>5$). from RProjects/PhDGraphs}
  \label{table:gamma}
\end{table}

\subsection{RESULTS Assortativity and methods degree assortativity}
\textcolor{red}{Move to discussionj}We can compare the average degree with those found in other publications. We can also look at the assortativity between nodes for degree. The PSP networks shows evidence of some negative degree assortativity (-0.18) using the commands \texttt{assortativity(g,deg,directed=FALSE)} in igraph for R \todo{Assortativity for essential and id}.

The degree association is negative in most real world networks other than social networks \cite{newman2002assortative}. \todo{include the peel multi-scale mixing patterns} see table \ref{Table:DegreeAssortativityNewman}\todo{? add positive $r$ to table}. Both the Barabsi Albert model and random model have 0 correlation. The Callaway model of network generation ('grown graph' - newman) has a positive degree correlation consistent with older nodes being connected to each other and hence in turn being more likely to be connected to others. This is at odds with our later finding (see orthologues) that nodes with clearly identified yeast orthologues tend to be of higher degree (this would fit both the Barabasi Albert model and Callaway) \todo{Do i add the pseudo hub bit here this is also probably good for the paper with Keith}. 

The assortativity suggests that the Barabasi Albert model is 'incomplete' as a model of the internet \cite{newman2002assortative} and in this network we also find evidence that our synaptic protein protein interaction has similar degree assortativity to that calculated by Newman which is of the yeast interactome (cite Jeong) . The reasons for this are not clear, there are other things in our study that support the Barabasi model and it may be that multiple processes are taking place. \todo{? move this to discussion}

The value of $r$ of -0.18 \todo{three sf} is similar to those cited by Newman \cite{newman2002assortative} in table \ref{Table:DegreeAssortativityNewman}. 

Although we expect the assortativity to be 0 in a random graph of the configuration model (the edges are placed at random so we would not expect there to be any association with edges and the degree of vertices), there will nevertheless be a deviation from 0 due to chance and it is helpful to quantify the assortativity by calculating this. 

The assortativity using a configuration model \todo{explain configuration model} is approximately normal centred on 0.
\textcolor{red}{table}
For 10,000 samples   

Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-2.142e-02 -3.811e-03 -1.743e-05  5.524e-05  3.861e-03  2.235e-02 

Shapiro-Wilk normality test

data:  results[1:500]
W = 0.99538, p-value = 0.1446


\begin{table}[]
    \centering
    \begin{tabular}{c|c|c}
       network  &N& $r$  \\
       \hline
       internet & 10697&-0.189\\
       world wide web &269504 & -0.065\\
       post synaptic proteome & 3457 & -0.18\\
       protein interactions & 2115 & -0.156\\
       neural network & 307 & -0.163\\
       food web & 92 & -0.276 \\
       
       
         
    \end{tabular}
    \caption{Degree assortativity coefficients of networks adapted from Newman 2002 \cite{newman2002assortative} including post synaptic proteome}
    \label{Table:DegreeAssortativityNewman}
\end{table}

%\subsubsection{Degree correlation}
 
\todo{Include \url{R_plot_degree_correlation.png} commented out}


Newman \cite{newman2018networks} p336 cites Maslov \cite{maslov2004detection} as giving as a reason for the disassortative nature of many empirical networks the prohibition on multi-edges. The expected number of multiedges between nodes would be greater than one (need to look at the paper again).

This is possibly the bit where Keith's bit would be good to put in.  
 %\begin{figure}]h]
 % \includegraphics[width=\linewidth]{Rplot_degree_correlation.png}
 % \caption{Plot of degree with k nearest neighbour mean of degree}
 % \label{fig:knn_degree}
%\end{figure}
%
%
%

\subsection{Assortativity with gene score}
 
A second way in which we could see if genes within the PSP that are enriched for genetic differences in education and intelligence globally within the PSP would be to calculate the correlation between there being and edge between two vertices and the significance score of the two vertices. A positive result would suggest that more significant (avoiding a cutoff effect) genes have protein products that are more likely to interact in the PSP graph. We can do this by calculating the assortativity as described by Newman. As a measure of significance we use the z score for each gene in the PSP for each study as provided by MAGMA. As some genes in the PSP do not have a significant number of SNPs within them they do not have a Z score as assigned by magma and these are therefore set to 0. \footnote{code \url{source('~/RProjects/centrality/R/get_gene_scores/assortativity_with_z_scores.R')} }

\subsubsection{Assortativity with z score results}
 
 \begin{table}[]
     \centering
     \begin{tabular}{ccc}
     \toprule
         Study & Missing values  & Assortativity\\
         \midrule
         Intelligence\textsubscript{Replication} & 148 & 0.003\\
         Education\textsubscript{Replication} & 172 & -0.006\\
         Education\textsubscript{Discovery} & 156 & 0.001\\
         Intelligence\textsubscript{Discovery} & 156 & -0.004\\
         \bottomrule
     \end{tabular}
     \caption{Assortativity of PSP graph vertices by study Z score. PSP graph members without corresponding Z score marked as missing values had z set to 0.}
     \label{tab:Assortativity of PSP graph and z scores}
 \end{table}


The results are shown in table\ref{tab:Assortativity of PSP graph and z scores}. Assortativity scores tend to me modest (as score of approximately 0.15 is often found for degree assortativity). These results are close to zero. 

This suggests that nodes with high or low z scores in the population cohorts are no more likely to have links to one another \textit{globally} across the network. It does not state that there cannot be areas within the graph where there are areas of high assortativity or areas where one high score connects to linker node connects to another high score in a group.

\todo{do this with pli too and for a phenotype eg GO glutamate}
\section{Association between significant vertices}
Do there tend to be edges between significant vertices? Do the vertices representing genes of genome wide significance form an induced subgraph that is connected?
The induced subgraph \todo{define} of significant genes for Intelligence\textsubscript{Replication} is 16 vertices no edges. Simulation mean 0.60 sd 0.94 N iterations = 1000. For Education\textsubscript{Replication} it is 25 vertices no edges (simulation mean 1.51 sd 1.69). For Intelligence \textsubscript{Discovery} it is 58 vertices with 4 edges. Simulation mean 8.78 sd 5.13. For Education \textsubscript{Discovery} it is 51 vertices with 2 edges. Simulation mean 6.3 SD 4.1.

Simulation of joint results out of 1000, 100 trials
Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    4.00    6.00    5.98    8.00   14.00 
   
   max = 14/1000 = 0.014
   mean = 0.006 of getting this number of edges or fewer in 4 samples of random subgraphs eg(number of times subgraph is 0 and next is 0 or less and next is 4 or less and next is 51 or less given actual frequency 0,0,4,2 and number of samples 16,25,58,51.
   
 


 
 \todo{Assortativity of essential nodes in PSP}
 \todo{difference plI and DEG}
 
 \subsection{Results from database of essetnail genes}

2241 genes in the PSP are essential (64.8\%)

This compares with 37.\% of the genome as a whole (assuming 20000 genes \todo{get non essential number from database}. 36.5\% (6043/20000) non PSP genes are essential. 

\subsubsection{Essential gene conclusion}
PSP genes are more likely to be essential

\todo{Degree etc for essential genes}
\todo{Essential genes enrichment vs non essential PSP add to community detection section}
\todo{Essential gene count for significant genes vs psp genes}

 \subsubsection{Degree distribution of PSP genes - murine model}
 
 We can use the mouse phenotype ontology to identify essential murine genes. 
 We want to know if high degree PSP nodes are more commonly associated with essentiality in murine models. 
 
 The 90\% centile for PSP gene degree is $k=37$. 349 genes in the PSP network model have degree $k>=37$.
 
 Murine phenotype MP:008763 is embryonic lethality. 117 of the genes with degree $k>=37$ are essential (33.5\% of 349 genes). 458 genes are essential in the remaining PSP nodes ($k < 37$, 458 of 3108 14.7\%). p $3.13 \times 10^{-11}$ Fisher's exact test for count data R. Odds ratio 2.27 95\% CI 1.79-2.88. \footnote{code:\url{/home/grant/RProjects/phenome_and_network/R/get_degree_90_quantile_PSP.R}} \todo{Calculate difference in degree for essential versus not essential} Comment \todo{one reason for testing the top 10\% is that the degree distribution is of a few hubs (cite Jeong) with different properties from the rest given the scale free structure of the network}
 
 \begin{table}[h]
     \centering
     \begin{tabular}{llllll}
          Title & $k$& n essential & n & percentage essential & p   \\
          \hline
          Top 10\% & $>=37$ & 117 & 349 & 33.5 & $3.13 \times 10^{-11}$\\
          Rest & $<37$ & 458 & 3108 & 14.7 & \\
     \end{tabular}
     \caption{Association of high degree nodes with being essential in murine models. Genes with degree in top 10\% of PSP compared with the rest. P Fisher's Exact Test for Count data.}
     \label{Table:Degree and murine essentialness PSP}
 \end{table}
 

\todo{GO enrichment with background PSP for high degree are in \url{/home/grant/Dropbox/stront_share/data/GO\_high\_degree_psp_background}}




\subsection{Degree and orthologs}

If the network is generated by the preferential attachment model of Barabasi we would expect the nodes of high degree to have more ancient orthologues than those of lower degree.  Duplication divergence \todo{ref} may be an important mechanism as we know there was a genome duplication event nevertheless we could hypothesise that the nodes with yeast orthologues will have higher median degree than the human and murine PSP. \todo{add cross ref to orthologues}

\section{Probability of loss of function intolerance (pLI)}

\todo{Remove takehome} \subsubsection{Takehome: Remove}

The probability of loss of function intolerance is associated with measures of vertex centrality in the post synaptic proteome graph. The correlation is greatest for degree ($\rho=0.230$) and lowest for transitivity ($\rho=0.086$). Betweenness and eigenvector centrality have similar correlation to loss of function intolerance as degree (see table \ref{Table:Correlation of pLI with centrality vertex statistic})



\subsection{Distribution of pLI and intelligence}
\todo{Distribution of pLI and intelligence}


% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 15:09:57 2020
\begin{table}[ht]
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{rllrrrl}
  \hline
 & Study & Graph statistic & S & p & rho & Test \\ 
  \hline
1 & Intelligence Discovery & pLI & $5.400 \times 10^{9}$ & $1.10 \times 10^{-6}$ & -0.087 & Spearman's rank correlation rho \\ 
  2 & Intelligence Replication & pLI & $5.200 \times 10^{9}$ & $1.07 \times 10^{-2}$ & -0.046 & Spearman's rank correlation rho \\ 
  3 & Education Discovery & pLI & $5.600 \times 10^{9}$ & $1.38 \times 10^{-11}$ & -0.120 & Spearman's rank correlation rho \\ 
  4 & Education Replication & pLI & $5.200 \times 10^{9}$ & $1.54 \times 10^{-4}$ & -0.068 & Spearman's rank correlation rho \\ 
   \hline
\end{tabular}
\end{adjustbox}
\caption{Correlation of GWAS gene level statistics Spearman's rho gene pLI} 
\label{Table:Correlation of GWAS gene level statistics Spearmans rho gene pLI}
\end{table}

% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Fri Mar  6 15:52:12 2020
\begin{table}[ht]
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{llrrr}
  \hline
study & comparison & rho & S & p \\ 
  \hline
Intelligence Replication & all Zstat and pLI & 0.043 & 797235155436.9 & $2.358 \times 10^{-8}$ \\ 
  Intelligence Replication & synaptic Zstat and pLI & 0.042 & 5465708502.3 & $1.661 \times 10^{-2}$ \\ 
  Intelligence Replication & non synaptic Zstat and pLI & 0.033 & 428265764841.3 & $1.249 \times 10^{-4}$ \\ 
  Education Replication & all Zstat and pLI & 0.065 & 749149170637.4 & $4.370 \times 10^{-17}$ \\ 
  Education Replication & synaptic Zstat and pLI & 0.070 & 5198989756.7 & $6.926 \times 10^{-5}$ \\ 
  Education Replication & non synaptic Zstat and pLI & 0.059 & 398992399833.6 & $6.604 \times 10^{-12}$ \\ 
  Intelligence Discovery & all Zstat and pLI & 0.066 & 778515153123.3 & $8.384 \times 10^{-18}$ \\ 
  Intelligence Discovery & synaptic Zstat and pLI & 0.078 & 5218943442.6 & $7.746 \times 10^{-6}$ \\ 
  Intelligence Discovery & non synaptic Zstat and pLI & 0.048 & 422374139606.0 & $1.396 \times 10^{-8}$ \\ 
  Education Discovery & all Zstat and pLI & 0.093 & 755468138230.8 & $2.195 \times 10^{-34}$ \\ 
  Education Discovery & synaptic Zstat and pLI & 0.117 & 5003003452.8 & $2.793 \times 10^{-11}$ \\ 
  Education Discovery & non synaptic Zstat and pLI & 0.071 & 412373757794.5 & $7.820 \times 10^{-17}$ \\ 
   \hline
\end{tabular}
\end{adjustbox}
\caption{Extra table to be removed shows spearman for all synaptic and non synaptic and all but rho is with pLI and zstat not -log10 and -log10 of p  so rho is positive ie}
\label{tab:Extra table to be removed shows spearman for all synaptic and non synaptic and all but rho is with pLI and zstat not -log10 and -log10 of p  so rho is positive ie}
\end{table}

There was a weak association between probability of loss of function intolerance and gene score in this case z score. In table \ref{Table:Correlation of GWAS gene level statistics Spearmans rho gene pLI} the correlation is shown with the negative log10 transform of the p value and so $\rho$ is negative. Fuller results are shown in table ~\ref{tab:Extra table to be removed shows spearman for all synaptic and non synaptic and all but rho is with pLI and zstat not -log10 and -log10 of p  so rho is positive ie}  \footnote{\url{source('~/RProjects/paper_xls_output/R/make_df_correlation_pLI_genescore.R')}} the relationship is stronger for synaptic genes than for non synaptic genes although the effect is small and synaptic genes as shown below have a higher probability of loss of function intolerance.
\todo{visualisation or graph}
There was a positive correlation between loss of function intolerance and the gene Z stat. To be explicit the Z stat is high and positive when the gene has a more significant association with the phenotype and hence a lower p value. There is a weak association between high loss of function intolerance and significant (low p value genes)

\todo{there is something wrong here as Jan we find correlation} 
 In testing correlation spearman’s rank correlation was used as the distribution of the probability of loss of function intolerance statistic is very non linear. It has a bi-modal distribution (see figure~\ref{fig:density estimate pLi} with a concentration of probability mass around 0 and 1 .


The distribution of probabilitity of loss of function intolerance in non PSP genes is still bimodal but the probability mass around 1 is much less peaked and there is greater mass at the bottom of the distribution (see figure~\ref{fig:density estimate pLi two panel}. The boxplot in figure ~\ref{fig:Boxplot of pli} shows the median and means of the distributions\footnote{ Source for graphs \url{source('~/RProjects/paper_xls_output/R/plot_pli_PSP_v_restgenome.R')}}. 


\begin{figure}
    \centering
    \includegraphics[width=0.9\textwidth]{images/Rplot_kernel_density.png}
    \caption{The distribution of probability of loss of function intolerance. Kernel density estimate using Guassian kernel. Bimodal distribution with peak close to 0 and 1 for the post synaptic proteome. }
    \label{fig:density estimate pLi}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.9\textwidth]{images/Rplot01_density_PSP_two_panel_nonPSP.png}
    \caption{The distribution of probability of loss of function intolerance for the post synaptic proteome (left) and the rest of the genome (post synaptic proteome removed) on the right. Kernel density estimate using Guassian kernel. Bimodal distribution with peak close to 0 and 1 for the post synaptic proteome. The peak around 1 is much smaller in the non post synaptic proteome. }
    \label{fig:density estimate pLi two panel}
\end{figure}



This code \url{source('~/RProjects/paper_xls_output/R/make_pLI_correlation.R')}
makes the table for the correlation between pLI and vertex statistics. It is the only one found in the previous hard copy so suggest \todo{Redo from scratch correlation between pLI and study p vals to confirm although findings makes sense and cannot inf any previous source code}.

The current results suggest that the relationship between the variables is as shown in the factor graph in figure \ref{Figure:Factor graph for relationship between pLI, vertex statistic and study p values}

\begin{figure}
    \centering
    \includegraphics[width=0.9\textwidth]{images/Rplot03_boxplot_pLI_PSP_non_PSP.png}
    \caption{Box plot of probability of loss of function intolerance (pLI) between PSP genes (n=) and non PSP genes (rest of genome n=14990 ). Diamond marker indicates mean PSP value. }
    \label{fig:Boxplot of pli}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{IMG_6979.JPG}
    \caption{Factor graph for relationship between pLI, vertex statistic and study p values}
    \label{Figure:Factor graph for relationship between pLI, vertex statistic and study p values}
\end{figure}
\todo{Make into table}
% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 15:52:39 2020
\begin{table}[ht]
\centering
\begin{tabular}{rllrrrl}
  \hline
 & Study & Graph statistic & S & p & cor & Test \\ 
  \hline
1 & pLI & Degree & $4.300 \times 10^{9}$ & $2.00 \times 10^{-40}$ & 0.230 & Spearman's rank correlation rho \\ 
  2 & pLI & Betweenness & $4.500 \times 10^{9}$ & $1.40 \times 10^{-32}$ & 0.210 & Spearman's rank correlation rho \\ 
  3 & pLI & Eigenvector & $4.400 \times 10^{9}$ & $6.60 \times 10^{-37}$ & 0.220 & Spearman's rank correlation rho \\ 
  4 & pLI & Transitivity & $4.000 \times 10^{9}$ & $2.80 \times 10^{-6}$ & 0.086 & Spearman's rank correlation rho \\ 
   \hline
\end{tabular}
\caption{Correlation of pLI with centrality vertex statistic. Code \url{source('~/RProjects/paper_xls_output/R/get_exac_correlation.R')}} 
\label{Table:Correlation of pLI with centrality vertex statistic}
\end{table}
\subsubsection{pLI and vertex statistics}
 The   correlation between pLI and vertex statistics is shown in table table \ref{Table:Correlation of pLI with centrality vertex statistic} \footnote{ code at \url{source('~/RProjects/paper_xls_output/R/make_pLI_correlation.R')}}. This suggests that a correlation between centrality measure and loss of function intolerance where more important genes in terms of centrality measure are more likely to be intolerant of genetic change. The correlation is greatest for degree ($\rho = 0.23$) but is similar for betweenness centality and eigenvector centrality. The correlation is much weaker for transitivity. 
 
 
\subsubsection{Linear model of pLI and vertex statistics}    
A linear regression model for centrality measures and pLI was constructed \footnote{Code \url{source('~/RProjects/paper_xls_output/R/make_pLI_correlation.R')}}. Although the correlation may be similarly linear the eigenvector centrality may have more effect on the gradient of the line of the relationship. 

The eigenvector centrality seems to predict pLI best. Taking the 3rd quartile of the eigenvector centrality then the following are the summary statistics for the pLI

The genes with higher eigenvector centrality tend to have higher probability of loss of function intolerance see table~\ref{tab:pli and eigenvector centrality quartiles}

\begin{table}[]
    \centering
    \begin{tabular}{llllllll}
    \toprule
      Sample &  Min. &1st Qu.&  Median &   Mean& 3rd Qu.&    Max. &   NA's     \\
      \midrule
     Eigenvector 3rd quartile  &0& 0.349& 0.9315& 0.689& 0.998& 1&     48  \\ 
     PSP &0&0.007&0.620&0.522&0.987&  1 & 222\\
     Eigenvector less than 3rd quartile &0&0.002&0.386&0.466&0.974&1&    174\\
     \bottomrule
    \end{tabular}
    \caption{Probability of loss of function intolerance for different values of eigenvector centrality and for PSP as whole}
    \label{tab:pli and eigenvector centrality quartiles}
\end{table}
%   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
% 0.0000  0.3485  0.9315  0.6892  0.9983  1.0000      48 

%For the PSP as a whole this is	
%  Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
%0.00000 0.00719 0.61952 0.52215 0.98691 1.00000     222 

%and for the eigenvector centrality below the third quartile it is

%  Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
%0.00000 0.00165 0.38598 0.46570 0.97380 1.00000     174

Wilcoxon test for difference in means for all synaptic pLI and those with eigenvector centrality above the third quartile
        Wilcoxon rank sum test with continuity correction

data:  df\$pLI and joined\_df\_synaptic\$pLI
W = 1611226, p-value $< 2.2e-16$
alternative hypothesis: true location shift is not equal to 0


\subsection{pLI for orthologs} \url{source('~/RProjects/orthologs2/R/make_PLI_orthologs.R')}

\todo{Unite this bit with the other ortholog bit}
\todo{Check if this is synaptic and add number}
\begin{table}[ht]
    \centering
    \begin{tabular}{lcccccc}
    &                 Min.& 1st Qu. &Median& Mean& 3rd Qu.& Max.\\
yeast\_sum   &     1.3e-23& 2.5e-02&  0.710& 0.56&    0.98 &   1\\
cel\_sum     &     2.9e-40& 1.3e-02&  0.630& 0.52&    0.97 &  1\\
fly\_sum      &    9.6e-51 &1.9e-02&  0.720& 0.56&    0.99 &  1\\
zf\_sum        &   9.6e-51& 1.1e-02&  0.670& 0.53&    0.99  &  1\\
mouse\_pLI      &  5.4e-91 &6.0e-03&  0.620& 0.52&    0.99   &1\\
pli\_synaptic\_sum & 5.4e-91 &7.2e-03&  0.620& 0.52&    0.99  &  1\\
human\_genome\_plI &5.4e-91 &3.2e-05&  0.028& 0.30&    0.68  &  1\\
  
    \end{tabular}
    \caption{Distribution of pLI (probability of loss of function intolerance orthologs}
    \label{tab:pLI orthologs}
\end{table}



    
\begin{figure}
    \centering
    \begin{subfigure}[t]{0.45\textwidth}
        \centering
        \includegraphics[width=\linewidth]{images/Rplot_transitivity.png} 
        \caption{Histogram of transitivity} \label{fig:transitivity}
    \end{subfigure}
    \hfill
    \begin{subfigure}[t]{0.45\textwidth}
        \centering
        \includegraphics[width=\linewidth]{images/Rplot01_logdegree_log_transitivity.png} 
        \caption{Plot of degree with mean local transitivity. Log10-log10 scale. Omitting degree $k=1$ as transitivity $C$ will be undefined.} \label{fig:log_transitivity_degree}
    \end{subfigure}
    \caption{Plots of transitivity \textcolor{red}{may want to do this in subfigure packages rather than subcaption to line up edges}}
\end{figure}



%\begin{figure}]h]
%  \includegraphics[width=\linewidth]{Rplot_transitivity}
%  \caption{Transitivity}
%  \label{fig:transitivity}
%\end{figure}
 
%\begin{figure}]h]
%  \includegraphics[width=\linewidth]{Rplot_transitivity_degree}
%  \caption{Local clustering coefficient and degree}
%  \label{fig:transitivity_degree}
%\end{figure}


%\begin{figure}
%    \centering
%    \includegraphics[width=\linewidth]{images/Rplot01_logdegree_log_transitivity.png}
%    \caption{Plot of degree with mean local transitivity. Log10-log10 scale. Omitting degree $k=1$ as transitivity $C$ will be undefined.}
 %   \label{fig:log_transitivity_degree}
%\end{figure}






\section{RESULTS: Centrality measures and graph statistics and Intelligence}

The correlation between the graph vertex centrality measures eigenvector centrality, betweenness centrality and transitivity and GWAS gene scores are shown in table \ref{Table:Correlation of GWAS gene level statistics with graph vertex measures}

The correlation between the graph vertex centrality measure statistics and the -log \textsubscript{10} transform of the vertex study p value is shown in table \ref{Table:Correlation of GWAS gene level statistics -log10 transform with graph vertex measures}

The Spearman rank correlation for -log\textsubscript{10} is shown in table \ref{Table:Correlation of GWAS gene level statistics Spearmans rho -log10 transform with graph vertex measures} and the rank correlation for p value and vertex statistic is found in table \ref{Table:Correlation of GWAS gene level statistics Spearmans rho gene p  with graph vertex measures} \todo{change S to scientific notation}

The vertex statistics are not normally distributed \todo{test this} and so we prefer the Spearman's rho. the -log\textsubscript{10} has the same value as $-\rho$ for p (signs switch).

\todo{Add multiple testing correction}


 \subsection{k coreness and GWAS results}
I also calculate the correlation between p and the coreness value of the vertex which is its k core \todo{expand}. There was no significant correlation as shown in table \ref{Table:Correlation of GWAS gene level statistics Spearmans rho gene p  with kcorecoreness measures}

%Code to complete this table is found at STRONTIUM\url{source('~/RProjects/paper_xls_output/R/make_df_graph_stats_correlation_PhDlatex.R')}.
\todo{fixed now}
% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 12:31:46 2020
\begin{table}[ht]
\centering
\begin{adjustbox}{max width=\textwidth}
\begin{tabular}{rllrlrrrrl}
  \hline
 & Study & Graph statistic & t & df & p & cor & CI lower & CI upper & Test \\ 
  \hline
1 & Intelligence Discovery & Degree & 0.75 & 3299 & 0.45 & 0.01 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  2 & Intelligence Discovery & Betweenness & 0.12 & 3299 & 0.90 & 0.00 & -0.03 & 0.04 & Pearson's product-moment correlation \\ 
  3 & Intelligence Discovery & Eigenvector & 0.73 & 3299 & 0.47 & 0.01 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  4 & Intelligence Discovery & Transitivity & -0.94 & 3012 & 0.35 & -0.02 & -0.05 & 0.02 & Pearson's product-moment correlation \\ 
  5 & Intelligence Replication & Degree & -0.64 & 3307 & 0.52 & -0.01 & -0.04 & 0.02 & Pearson's product-moment correlation \\ 
  6 & Intelligence Replication & Betweenness & -0.38 & 3307 & 0.71 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  7 & Intelligence Replication & Eigenvector & -0.33 & 3307 & 0.74 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  8 & Intelligence Replication & Transitivity & 1.00 & 3019 & 0.31 & 0.02 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  9 & Education Discovery & Degree & -0.25 & 3299 & 0.80 & -0.00 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  10 & Education Discovery & Betweenness & -1.10 & 3299 & 0.29 & -0.02 & -0.05 & 0.02 & Pearson's product-moment correlation \\ 
  11 & Education Discovery & Eigenvector & 1.00 & 3299 & 0.32 & 0.02 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  12 & Education Discovery & Transitivity & 0.09 & 3012 & 0.93 & 0.00 & -0.03 & 0.04 & Pearson's product-moment correlation \\ 
  13 & Education Replication & Degree & -1.20 & 3283 & 0.23 & -0.02 & -0.06 & 0.01 & Pearson's product-moment correlation \\ 
  14 & Education Replication & Betweenness & -0.63 & 3283 & 0.53 & -0.01 & -0.04 & 0.02 & Pearson's product-moment correlation \\ 
  15 & Education Replication & Eigenvector & -0.45 & 3283 & 0.65 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  16 & Education Replication & Transitivity & 1.50 & 2998 & 0.14 & 0.03 & -0.01 & 0.06 & Pearson's product-moment correlation \\ 
   \hline
\end{tabular}
\end{adjustbox}
\caption{Correlation of GWAS gene level statistics with graph vertex measures.Code to complete this table is found at STRONTIUM\url{source('~/RProjects/paper_xls_output/R/make_df_graph_stats_correlation_PhDlatex.R')}}
\label{Table:Correlation of GWAS gene level statistics with graph vertex measures}
\end{table}

% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 12:43:57 2020
\begin{table}[ht]
\centering
  \begin{adjustbox}{max width=\textwidth}
\begin{tabular}{rllrlrrrrl}
  \hline

 & Study & Graph statistic & t & df & p & cor & confidence\_int\_lower & condidence\_upper & Test \\ 
  \hline
1 & Intelligence Discovery & Degree & -0.40 & 3299 & 0.69 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  2 & Intelligence Discovery & Betweenness & 0.04 & 3299 & 0.97 & 0.00 & -0.03 & 0.04 & Pearson's product-moment correlation \\ 
  3 & Intelligence Discovery & Eigenvector & -0.31 & 3299 & 0.76 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  4 & Intelligence Discovery & Transitivity & 0.59 & 3012 & 0.56 & 0.01 & -0.03 & 0.05 & Pearson's product-moment correlation \\ 
  5 & Intelligence Replication & Degree & -0.14 & 3307 & 0.89 & -0.00 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  6 & Intelligence Replication & Betweenness & -0.41 & 3307 & 0.68 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  7 & Intelligence Replication & Eigenvector & -0.51 & 3307 & 0.61 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  8 & Intelligence Replication & Transitivity & -0.88 & 3019 & 0.38 & -0.02 & -0.05 & 0.02 & Pearson's product-moment correlation \\ 
  9 & Education Discovery & Degree & -0.84 & 3299 & 0.40 & -0.01 & -0.05 & 0.02 & Pearson's product-moment correlation \\ 
  10 & Education Discovery & Betweenness & -0.55 & 3299 & 0.58 & -0.01 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  11 & Education Discovery & Eigenvector & -1.70 & 3299 & 0.08 & -0.03 & -0.06 & 0.00 & Pearson's product-moment correlation \\ 
  12 & Education Discovery & Transitivity & 0.74 & 3012 & 0.46 & 0.01 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  13 & Education Replication & Degree & 1.10 & 3283 & 0.29 & 0.02 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  14 & Education Replication & Betweenness & 0.87 & 3283 & 0.38 & 0.01 & -0.02 & 0.05 & Pearson's product-moment correlation \\ 
  15 & Education Replication & Eigenvector & -0.05 & 3283 & 0.96 & -0.00 & -0.04 & 0.03 & Pearson's product-moment correlation \\ 
  16 & Education Replication & Transitivity & -2.00 & 2998 & 0.04 & -0.04 & -0.07 & -0.00 & Pearson's product-moment correlation \\ 
   \hline
\end{tabular}
\end{adjustbox}
\caption{Correlation of GWAS gene level statistics -log10 transform with graph vertex measures} 
\label{Table:Correlation of GWAS gene level statistics -log10 transform with graph vertex measures}
\end{table}


% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 12:58:15 2020
\begin{table}[ht]
\centering
  \begin{adjustbox}{max width=\textwidth}
\begin{tabular}{rllrrrl}
  \hline
 & Study & Graph statistic & S & p & rho & Test \\ 
  \hline
1 & Intelligence Discovery & Degree & 6200000000.00 & 0.04 & -0.04 & Spearman's rank correlation rho \\ 
  2 & Intelligence Discovery & Betweenness & 6200000000.00 & 0.05 & -0.03 & Spearman's rank correlation rho \\ 
  3 & Intelligence Discovery & Eigenvector & 6200000000.00 & 0.11 & -0.03 & Spearman's rank correlation rho \\ 
  4 & Intelligence Discovery & Transitivity & 4500000000.00 & 0.85 & 0.00 & Spearman's rank correlation rho \\ 
  5 & Intelligence Replication & Degree & 5900000000.00 & 0.20 & 0.02 & Spearman's rank correlation rho \\ 
  6 & Intelligence Replication & Betweenness & 5900000000.00 & 0.21 & 0.02 & Spearman's rank correlation rho \\ 
  7 & Intelligence Replication & Eigenvector & 6000000000.00 & 0.93 & -0.00 & Spearman's rank correlation rho \\ 
  8 & Intelligence Replication & Transitivity & 4700000000.00 & 0.34 & -0.02 & Spearman's rank correlation rho \\ 
  9 & Education Discovery & Degree & 6000000000.00 & 0.69 & -0.01 & Spearman's rank correlation rho \\ 
  10 & Education Discovery & Betweenness & 6000000000.00 & 0.92 & -0.00 & Spearman's rank correlation rho \\ 
  11 & Education Discovery & Eigenvector & 6200000000.00 & 0.07 & -0.03 & Spearman's rank correlation rho \\ 
  12 & Education Discovery & Transitivity & 4600000000.00 & 0.77 & -0.01 & Spearman's rank correlation rho \\ 
  13 & Education Replication & Degree & 5800000000.00 & 0.53 & 0.01 & Spearman's rank correlation rho \\ 
  14 & Education Replication & Betweenness & 5800000000.00 & 0.39 & 0.01 & Spearman's rank correlation rho \\ 
  15 & Education Replication & Eigenvector & 6000000000.00 & 0.65 & -0.01 & Spearman's rank correlation rho \\ 
  16 & Education Replication & Transitivity & 4600000000.00 & 0.10 & -0.03 & Spearman's rank correlation rho \\ 
   \hline
\end{tabular}
\end{adjustbox}
\caption{Correlation of GWAS gene level statistics Spearmans rho -log10 transform with graph vertex measures \url{source('~/RProjects/paper_xls_output/R/make_df_graph_stats_correlation_spearman_log10_PhDlatex.R')}} 
\label{Table:Correlation of GWAS gene level statistics Spearmans rho -log10 transform with graph vertex measures}
\end{table}

% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 13:06:00 2020
\begin{table}[ht]
\centering
  \begin{adjustbox}{max width=\textwidth}
\begin{tabular}{rllrrrl}
  \hline
 & Study & Graph statistic & S & p & rho & Test \\ 
  \hline
1 & Intelligence Discovery & Degree & 5800000000.00 & 0.04 & 0.04 & Spearman's rank correlation rho \\ 
  2 & Intelligence Discovery & Betweenness & 5800000000.00 & 0.05 & 0.03 & Spearman's rank correlation rho \\ 
  3 & Intelligence Discovery & Eigenvector & 5800000000.00 & 0.11 & 0.03 & Spearman's rank correlation rho \\ 
  4 & Intelligence Discovery & Transitivity & 4600000000.00 & 0.85 & -0.00 & Spearman's rank correlation rho \\ 
  5 & Intelligence Replication & Degree & 6200000000.00 & 0.20 & -0.02 & Spearman's rank correlation rho \\ 
  6 & Intelligence Replication & Betweenness & 6200000000.00 & 0.21 & -0.02 & Spearman's rank correlation rho \\ 
  7 & Intelligence Replication & Eigenvector & 6000000000.00 & 0.93 & 0.00 & Spearman's rank correlation rho \\ 
  8 & Intelligence Replication & Transitivity & 4500000000.00 & 0.34 & 0.02 & Spearman's rank correlation rho \\ 
  9 & Education Discovery & Degree & 6000000000.00 & 0.69 & 0.01 & Spearman's rank correlation rho \\ 
  10 & Education Discovery & Betweenness & 6000000000.00 & 0.92 & 0.00 & Spearman's rank correlation rho \\ 
  11 & Education Discovery & Eigenvector & 5800000000.00 & 0.07 & 0.03 & Spearman's rank correlation rho \\ 
  12 & Education Discovery & Transitivity & 4500000000.00 & 0.77 & 0.01 & Spearman's rank correlation rho \\ 
  13 & Education Replication & Degree & 6000000000.00 & 0.53 & -0.01 & Spearman's rank correlation rho \\ 
  14 & Education Replication & Betweenness & 6000000000.00 & 0.39 & -0.01 & Spearman's rank correlation rho \\ 
  15 & Education Replication & Eigenvector & 5900000000.00 & 0.65 & 0.01 & Spearman's rank correlation rho \\ 
  16 & Education Replication & Transitivity & 4400000000.00 & 0.10 & 0.03 & Spearman's rank correlation rho \\ 
   \hline
\end{tabular}
  \end{adjustbox}
\caption{Correlation of GWAS gene level statistics Spearmans rho gene p with graph vertex measures \url{source('~/RProjects/paper_xls_output/R/make_df_graph_stats_correlation_spearman_P_PhDlatex.R')}} 
\label{Table:Correlation of GWAS gene level statistics Spearmans rho gene p  with graph vertex measures}
\end{table}

% latex table generated in R 3.6.2 by xtable 1.8-4 package
% Sat Jan 11 13:27:32 2020
\begin{table}[ht]
\centering
  \begin{adjustbox}{max width=\textwidth}
\begin{tabular}{rllrrrl}
  \hline
 & Study & Graph statistic & S & p & rho & Test \\ 
  \hline
5 & Intelligence Discovery & Coreness & 5800000000.00 & 0.07 & 0.03 & Spearman's rank correlation rho \\ 
  10 & Intelligence Replication & Coreness & 6200000000.00 & 0.27 & -0.02 & Spearman's rank correlation rho \\ 
  15 & Education Discovery & Coreness & 5900000000.00 & 0.52 & 0.01 & Spearman's rank correlation rho \\ 
  20 & Education Replication & Coreness & 5900000000.00 & 0.79 & -0.00 & Spearman's rank correlation rho \\ 
   \hline
\end{tabular}
\end{adjustbox}
\caption{Correlation of GWAS gene level statistics Spearmans rho gene p with kcore coreness measures \url{source('~/RProjects/paper_xls_output/R/make_df_graph_stats_correlation_spearman_P_kcore_PhDlatex.R')}} 
\label{Table:Correlation of GWAS gene level statistics Spearmans rho gene p  with kcorecoreness measures}
\end{table}






\subsection{Results Centrality and murine models of long term potentiation}
\label{sec:results centrality and murine models of long term potentiation}
    The centrality measures of genes marked as murine models of LTP were considered as model animal models of intelligence. \footnote{\url{source('~/RProjects/centrality/R/model_ltp/model ltp.R')}}.
    256 genes were identified as a
    ssociated with murine LTD from \textcolor{red}{add Mouse ontology}\todo{add mouse ontology}. 141 of these genes were found in the PSP network model (55.1\%).
    
    There was no statistically significant difference in association between the centrality measures of those genes associated with LTP and the rest of the PSP. Tested degree (0.809), eigenvector centrality (0.630), betweenness (0.895) and transitivity (0.898) Wilcoxon rank sum test with continuity correction. 
    
    A logistic regression model of these factors and LTP PSP as outcome showed that eigenvector centrality and degree were significant factors (4.04 $\times 10^{-6}$ and $1.39 \times 10^{-6}$. The effects were very minor however with McFadden's pseudo R2 \cite{mcfadden1973conditional} calculated using the pscl package was  0.031 \cite{jackman2017package}.

\subsubsection{Other notes}

enrichment of genes found in the synapse with orthologs in yeast reveal a large number for exosome and mitochondrial elements
\textbf{Articulation points}
11 of 68 glutamate receptor binding are articulation points however Gene ontology enrichment analysis ofarticulation points with the entire PSP as background yileds no significant over-representation of terms. They appear to be under represented in ID (3 genes of 203.	 	
\section{Summary}

We show that in the PSP nodes of very high centrality are associated with essentialness. The PSP has a high degree of essentialness compared to the rest of the genome. 

High degree nodes are associated with specific phenotypes and specific gene ontology terms however the are not correlated with genetic differences that are enriched for differences in intelligence. This is not obvious before the fact and is different from other traits eg essentialness. One might hypothesis that the reason is that there is either no effect (ie it is irrelevant) or that in population studies genes with high centrality are more likely to be essential and hence have less genetic variation. We find also in one of the principle animal models of learning murine long term potentiation that genes of high importance were no more likely or less likely to be involved in long term potentiation.

We use data from the Exac project to further investigate whether high importance nodes were subject to selection pressure such that they proved more intolerant of non synonymous genetic changes. Nodes with high degree are associated with an increased probability of being intolerant to changes. 

Nodes with high betweenness centrality were involved in neuro-degenerative disorders including Parkinson's disease see section~\ref{sec:Betweeness centrality}. They were also associated with severe phenotypes in murine models including phenotypes incompatibly with viability. Low betweenness centrality was associated with abnormalities in synaptic transmission in murine models. 

High eigenvector centrality is again associated with severe murine phenotypes and enrichment for RNA and cadherin binding. Disease enrichment is of neoplasms and neurodegenerative disorders. 

Low eigenvector centrality (10\%) were enriched for human phenotypes of seizures and ontology terms of transmitter, mouse phenotype was of abnormal synaptic transmission. The top five enriched disorders in disGeNET were seizure related. 

Despite this there was no association between node centrality measures and the significance of genetic variants for differences in intelligence and educational attainment.

There is a complex relationship between transitivity and degree for nodes with degree 3 or less the correlation is positive and results in a positive overall correlation despite a plot of the relationship appearing clearly negative. This is due to the large number of low degree nodes. When the correlation is calculated for higher degree the relationship previously commented upon in the literature \cite{newman} p 335 is seen. Caution should therefore be used in determining linear correlations involving transitivity especially in smaller networks and it may be prudent to plot the correlation against minimum degree.


 



